# GWAS Tools and Data Preprocessing Workflow with Toy Example -- PLINK and IMPUTE2

This notebook demonstrates a standard approach to conduct the genome wide association study (GWAS) using toy example datasets. The procedures consist of four steps:

- Quality Control
- Imputation
- Population Stratification Correction
- Simple Association Test
The data used for this example is a toy genotype data containing 5 random SNPs and a random phenotype data with covariates.

We use both bash and R codes in this notebook. The type of programming language is labeled at the top of each code cell.

::: {#note style="color: blue;"}
Special notes for Windows users: Windows has two command-line shells, Command shell and Windows Powershell. Command shell is specific to Windows system and can be run from RStudio Terminal or in "Command Prompt" App, but the language it uses is NOT the same as Bash! Powershell App is the only place to run Bash commands in Windows. Therefore if you are Windows users and use RStudio for this course, you can't run bash codes such as "head" or "awk" from RStudio Terminal, you can **ONLY** run PLINK program codes. If you would like to try Bash codes, please open "Windows Powershell" app and follow the same steps to set the paths. Still, Powershell App won't run PLINK codes, so you'll have to switch between Powershell and RStudio Terminal.
:::

## Environment Setting

Before we can run any analysis, we need to set the working directory path to the downloaded "bigcare-main" folder for both Terminal and Console in RStudio. Below are the codes to setup paths at both places. These paths have to be setup correctly at first, otherwise we can't locate the correct input files.

```{r eval=F}
# change the paths to where you saved "bigcare-main"
## run in Terminal
mydir="/path/to/bigcare-main"
cd $mydir

# R codes- run in R Console
setwd("/path/to/bigcare-main")
```

## GWAS Tools
We will use two main program tools in this notebook, PLINK and IMPUTE2.

PLINK is used for quality control and perform association test for large scale genotype data. You can download PLINK program using this link. Download the .zip file or find the .zip files suitable for your laptop system in "bigcare-main/Tools/PLINK" and unzip it to call the plink program in the folder.

IMPUTE2 is used for genotype imputation and haplotype phasing. You can download its program using this link. IMPUTE2 program is only available to run in Linux or macOS system. You can find the zipped installation file in "bigcare-main/Tools/IMPUTE2". If you want to perform imputation for your own data, you also need to download the required reference genome from the above link. But in this notebook will only use the example genome.

::: {#note style="color: blue;"}
Important: If you want to run these two programs on your laptop, please make sure you have unzipped the installation packages in "./bigcare-main/Tools" that is suitable for your laptop systerm. We will directly call PLINK and IMPUTE2 programs using relative path.
:::

Instructions on installing these two tools can be found on course GitHub here.

For example, my computer is macOS system so I can run both PLINK and IMPUTE2. After I downloaded "bigcare-main" folder, I open ./Tools/PLINK and unzip the plink package for macOS: "plink_mac_20230116.zip". A folder "plink_mac_20230116" will be created which contains a program file called "plink". Then, I can call PLINK program by specifying the absolute path of this "plink" program: /path/to/bigcare-main/Tools/PLINK/plink_mac_20230116/plink. Similar for Windows system, you need to unzip the package "plink_win64_20230116.zip" with right click->"Extract All...". Then you can call PLINK with /path/to/bigcare-main/Tools/PLINK/plink_win64_20230116/plink.exe.

Similar for IMPUTE2, I unzip "impute_v2.3.2_MacOSX_Intel.tgz" for macOS under "./bigcare-main/Tools/IMPUTE2/". And I can call IMPUTE2 program with /path/to/bigcare-main/Tools/IMPUTE2/impute_v2.3.2_MacOSX_Intel/impute2. Note that Windows system won't be able to run IMPUTE2.

You should modify the above /path/to/bigcare to where your "bigcare-main" folder is.

::: {#note style="color: blue;"}
However, macOS users need to go through additional steps to authorize PLINK and IMPUTE2 programs before you can call them! See below.
:::

When you try to call PLINK or IMPUTE2 with the above commands, you might get a pop-up window with the following message “plink” cannot be opened because the developer cannot be verified.. Select "Cancel" and we will authorize PLINK first. Please go to macOS "System Settings"->"Privacy & Security", and scroll down until you see a message "plink" was blocked from use because it is not from an identified developer.. Click "Allow Anyway" and you will be prompt to enter your Mac username and password. Click "Unlock" and then "Allow Anyway" again, and the previous message will disappear. Then try to run the same codes again in RStudio Terminal. You will get another pop-up window saying macOS cannot verify the developer of “plink”. Are you sure you want to open it?, select "Open" then you can see PLINK or IMPUTE2 commands options show up in Terminal.


## Quality Control
PLINK will generate a number of standard summary statistics that are useful for quality control (e.g. missing genotype rate, minor allele frequency, Hardy-Weinberg equilibrium test and non-Mendelian transmission rate). These can also be used as thresholds for subsequent analyses. For more detailed information about plink, please refer its website http://zzz.bwh.harvard.edu/plink/.

The following is a list of some common options that can be used to filter out individuals or SNPs on the basis of the summary statistic measures:

```{r table2, echo=FALSE, message=FALSE, warnings=FALSE, results='asis'}
tabl <- "
| Feature inclusion criteria |	As summary statistic | As inclusion criteria |
|----------------------------|-----------------------|-----------------------|
| Missingness per individual |`--missing`            |`--mind N`.            |
| Missingness per marker	   | --missing	           | --geno N              |
| Allele frequency	         | --freq	               | --maf N               |
| Hardy-Weinberg equilibrium |--hardy	               | --hwe N               |
"
cat(tabl) # output the table in a format good for HTML/PDF/docx conversion
```

## An Example of Summary Statistics
### Input files
Input files: in folder `./QC`-- We will use a random selected 5 SNPs as an example data:

- 5SNPs.ped
- 5SNPs.map

The PED file is a white-space delimited file: the first six columns are mandatory:

     Family ID
     Individual ID
     Paternal ID
     Maternal ID
     Sex (1=male; 2=female; other=unknown)
     Phenotype

```{bash}
# bash codes- run in Terminal
# Windows user need to run in Powershell App 
head ./Data/QC/5SNPs.ped
```
By default, each line of the MAP file describes a single marker and must contain exactly 4 columns:

     chromosome (1-22, X, Y or 0 if unplaced)
     rs# or snp identifier
     Genetic distance (morgans)
     Base-pair position (bp units)
```{bash}
head ./Data/QC/5SNPs.map
```
Three types of summary statistics that we are going to calculate using `PLINK`:

1. missingness checking
2. allele frequencies
3. hardy-weinberg

All results generated from the following commands are saved in folder `./Data/QC`. You can open the files in this folder to view the data structure.

### Calculating missing rate

The following commands compute the **missingness** for each SNP and each subject. (The output will print all calculation progress generated by `PLINK`.)

1. **--noweb**, suppress the connection to internet for version check.
2. **--file**, specify the input files.
3. **--missing**,  calculate the number of missing and missing rate.
4. **--out**, specify the output file name.

```{bash eval=F}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# for macoS users
./Tools/PLINK/plink_mac_20230116/plink \
--noweb --file ./Data/QC/5SNPs --missing --out ./Data/QC/younameit

# for windows users
./Tools/PLINK/plink_win64_20230116/plink.exe \
--noweb --file ./Data/QC/5SNPs --missing --out ./Data/QC/younameit
```

This option creates two output files in 'results' folder:

1. **"younameit.imiss"**, missingness by individual.
2. **"younameit.lmiss"**, missingness by SNP.

For individuals ".imiss" file, the format is:

     FID                Family ID
     IID                Individual ID
     MISS_PHENO         Missing phenotype? (Y/N)
     N_MISS             Number of missing SNPs
     N_GENO             Number of non-obligatory missing genotypes
     F_MISS             Proportion of missing SNPs
```{bash}
# bash codes- run in Terminal
# Windows user need to run in Powershell App 
head ./Data/QC/younameit.imiss
```

For SNP "lmiss" file, the format is:

     SNP                SNP identifier
     CHR                Chromosome number
     N_MISS             Number of individuals missing this SNP
     N_GENO             Number of non-obligatory missing genotypes
     F_MISS             Proportion of sample missing for this SNP

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App 
head ./Data/QC/younameit.lmiss
```

### Calculating allel frequencies

The following command computes the **allele frequencies**. (The output will print all calculation progress generated by `PLINK`.)
1. **--freq**, calculate the allele frequencies.

```{bash eval=F}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# macOS user
./Tools/PLINK/plink_mac_20230116/plink \
--noweb --file ./Data/QC/5SNPs --freq --out ./Data/QC/younameit

# Windows user
./Tools/PLINK/plink_win64_20230116/plink.exe \
--noweb --file ./Data/QC/5SNPs --freq --out ./Data/QC/younameit

```

This command will create a output file **"younameit.frq"** with six columns:

     CHR       Chromosome
     
     SNP       SNP identifier
     
     A1        Allele 1 code (minor allele)
     
     A2        Allele 2 code (major allele)
     
     MAF       Minor allele frequency
     
     NCHROBS   Non-missing allele count

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
head ./Data/QC/younameit.frq

```

