--- 
title: "2024 CSHL course - GWAS Hands-on Analysis"
author: "Danni Liu - Purdue University"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/cshl_2024
description: "This is the course material for the 2024 CSHL hands-on session in GWAS."
---

# Welcome to the GWAS and eQTL Workshop

Placeholder


### Overview
### Folder structure:
### Programming Languages: 
### Required GWAS Tools

<!--chapter:end:index.Rmd-->


# An Overview of Regression with Toy Examples

Placeholder


## Environment Setting and Packages
## Toy Example Data
## Logistic Regression
## ROC curve and AUC for HOXB13
## Practice
## Comparison of HOXB13 ROC curve and NHLH1 ROC curve
## Linear Regression
### Linear regression of HOXB13 vs. IL17BR

<!--chapter:end:01-intro.Rmd-->

# GWAS Tools and Data Preprocessing Workflow with Toy Example 

This notebook demonstrates a standard approach to conduct the genome wide association study (GWAS) using toy example datasets. The procedures consist of four steps:

- Quality Control
- Imputation
- Population Stratification Correction
- Simple Association Test
The data used for this example is a toy genotype data containing 5 random SNPs and a random phenotype data with covariates.

We use both bash and R codes in this notebook. The type of programming language is labeled at the top of each code cell.

> <span style="color: #0064a4;">  
Special notes for Windows users: Windows has two command-line shells, Command shell and Windows Powershell. Command shell is specific to Windows system and can be run from "Command Prompt" App, but the language it uses is NOT the same as Bash! Powershell App is the app to run Bash commands in Windows. Therefore if you are Windows users and use RStudio for this course, you can't run bash codes such as "head" or "awk" from RStudio Terminal, you can **ONLY** run PLINK program codes. If you would like to try Bash codes, please open "Windows Powershell" app and follow the same steps to set the paths. Still, Powershell App won't run PLINK codes, so you'll have to switch between Powershell and RStudio Terminal.
</span>

## Environment Setting

Before we can run any analysis, we need to set the working directory path to the downloaded "cshl_2024" folder for both Terminal and Console in RStudio. Below are the codes to setup paths at both places. These paths have to be setup correctly at first, otherwise we can't locate the correct input files.

```{r eval=F}
# change the paths to where you saved "cshl_2024"
## run in Terminal
mydir="/path/to/cshl_2024"
cd $mydir

# R codes- run in R Console
setwd("/path/to/cshl_2024")
```


## GWAS Tools
We will use two main program tools in this notebook, **PLINK** and **IMPUTE2**.

**PLINK** is used for quality control and perform association test for large scale genotype data. You can download PLINK program using this link. Download the .zip file or find the .zip files suitable for your laptop system in "cshl_2024/Tools/PLINK" and unzip it to call the plink program in the folder.

**IMPUTE2** is used for genotype imputation and haplotype phasing. You can download its program using this link. IMPUTE2 program is only available to run in Linux or macOS system. You can find the zipped installation file in "cshl_2024/Tools/IMPUTE2". If you want to perform imputation for your own data, you also need to download the required reference genome from the above link. But in this notebook will only use the example genome.

We will also introduce a tool called **MAGMA** for gene-set analysis in later chapters.

> <span style="color: #0064a4;">
Important: If you want to run these programs on your laptop, please make sure you have unzipped the installation packages in "./cshl_2024/Tools" that is suitable for your laptop systerm. We will directly call PLINK and IMPUTE2 programs using abusolute or relative paths.
</span>

Instructions on installing these two tools can be found on in Chapter 1.

After you cloned the repository "cshl_2024" folder, you can find the `Tools` folder contatining installation packages for different systems. Then, you can unzip the zip files and call the programs by specifying its absolute path. For example, for macOS users, you can unzip the installer by double clicking, and call PLINK using `/path/to/cshl_2024/Tools/PLINK/plink_mac/plink`; for Windows users, you need to unzip the package "plink_win64.zip" with right click->"Extract All...". Then you can call PLINK with `/path/to/cshl_2024/Tools/PLINK/plink_win64/plink.exe`.

You should modify the above `/path/to/cshl_2024` to where your "cshl_2024" folder is.

> <span style="color: #0064a4;">
However, macOS users need to go through additional steps to authorize PLINK, IMPUTE2 and MAGMA programs before you can call them! But you can perform these authorization steps only when you need to call the programs. See below.
</span>

When you try to call PLINK, IMPUTE2 and MAGMA with the above commands, you might get a pop-up window with the following message “plink” cannot be opened because the developer cannot be verified. Select "Cancel" and we will authorize PLINK first. Please go to macOS "System Settings"->"Privacy & Security", and scroll down until you see a message "plink" was blocked from use because it is not from an identified developer. Click "Allow Anyway" and you will be prompt to enter your Mac username and password. Click "Unlock" and then "Allow Anyway" again, and the previous message will disappear. Then try to run the same codes again in RStudio Terminal. You will get another pop-up window saying macOS cannot verify the developer of “plink”. Are you sure you want to open it?, select "Open" then you can see PLINK or IMPUTE2 commands options show up in Terminal.


## Quality Control
PLINK will generate a number of standard summary statistics that are useful for quality control (e.g. missing genotype rate, minor allele frequency, Hardy-Weinberg equilibrium test and non-Mendelian transmission rate). These can also be used as thresholds for subsequent analyses. For more detailed information about plink, please refer its website http://zzz.bwh.harvard.edu/plink/.

The following is a list of some common options that can be used to filter out individuals or SNPs on the basis of the summary statistic measures:

```{r table2, echo=FALSE, message=FALSE, warnings=FALSE, results='asis'}
tabl <- "
| Feature inclusion criteria |	As summary statistic | As inclusion criteria |
|----------------------------|-----------------------|-----------------------|
| Missingness per individual |`--missing`            |`--mind N`.            |
| Missingness per marker	   | --missing	           | --geno N              |
| Allele frequency	         | --freq	               | --maf N               |
| Hardy-Weinberg equilibrium |--hardy	               | --hwe N               |
"
cat(tabl) # output the table in a format good for HTML/PDF/docx conversion
```

## An Example of Summary Statistics
### Input files
Input files: in folder `./Data/QC`-- We will use a random selected 5 SNPs as an example data:

- 5SNPs.ped
- 5SNPs.map

The PED file is a white-space delimited file: the first six columns are mandatory:

     Family ID
     Individual ID
     Paternal ID
     Maternal ID
     Sex (1=male; 2=female; other=unknown)
     Phenotype

```{bash}
# bash codes- run in Terminal
# Windows user need to run in Powershell App 
head ./Data/QC/5SNPs.ped
```
By default, each line of the MAP file describes a single marker and must contain exactly 4 columns:

     chromosome (1-22, X, Y or 0 if unplaced)
     rs# or snp identifier
     Genetic distance (morgans)
     Base-pair position (bp units)
```{bash}
head ./Data/QC/5SNPs.map
```
Three types of summary statistics that we are going to calculate using `PLINK`:

1. missingness checking
2. allele frequencies
3. hardy-weinberg

All results generated from the following commands are saved in folder `./Data/QC`. You can open the files in this folder to view the data structure.

### Calculating missing rate

The following commands compute the **missingness** for each SNP and each subject. (The output will print all calculation progress generated by `PLINK`.)

1. **--noweb**, suppress the connection to internet for version check.
2. **--file**, specify the input files.
3. **--missing**,  calculate the number of missing and missing rate.
4. **--out**, specify the output file name.

```{bash eval=T}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# for windows users
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/QC/5SNPs --missing --out ./Data/QC/younameit

```

```{bash eval=F}
# for macoS users
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/QC/5SNPs --missing --out ./Data/QC/younameit

```

This option creates two output files in 'results' folder:

1. **"younameit.imiss"**, missingness by individual.
2. **"younameit.lmiss"**, missingness by SNP.

For individuals ".imiss" file, the format is:

     FID                Family ID
     IID                Individual ID
     MISS_PHENO         Missing phenotype? (Y/N)
     N_MISS             Number of missing SNPs
     N_GENO             Number of non-obligatory missing genotypes
     F_MISS             Proportion of missing SNPs

```{bash}
# bash codes- run in Terminal
# Windows user need to run in Powershell App 
head ./Data/QC/younameit.imiss
```

For SNP "lmiss" file, the format is:

     SNP                SNP identifier
     CHR                Chromosome number
     N_MISS             Number of individuals missing this SNP
     N_GENO             Number of non-obligatory missing genotypes
     F_MISS             Proportion of sample missing for this SNP

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App 
head ./Data/QC/younameit.lmiss
```

### Calculating allel frequencies

The following command computes the **allele frequencies**. (The output will print all calculation progress generated by `PLINK`.)
1. **--freq**, calculate the allele frequencies.

```{bash eval=T}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/QC/5SNPs --freq --out ./Data/QC/younameit
```
```{bash eval=F}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/QC/5SNPs --freq --out ./Data/QC/younameit
```

This command will create a output file **"younameit.frq"** with six columns:

     CHR       Chromosome
     
     SNP       SNP identifier
     
     A1        Allele 1 code (minor allele)
     
     A2        Allele 2 code (major allele)
     
     MAF       Minor allele frequency
     
     NCHROBS   Non-missing allele count

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
head ./Data/QC/younameit.frq

```

### Hardy-Weinberg Equilibrium
The following command computes the **Hardy-Weiberg test**. (The output will print all calculation progress generated by `PLINK`.)
1. **--hardy**, perform hardy-weinberg test.

```{bash eval=T}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/QC/5SNPs --hardy --out ./Data/QC/younameit
```
```{bash eval=F}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/QC/5SNPs --hardy --out ./Data/QC/younameit
```

This command will create an output file **"younameit.hwe"** with the following columns:

     SNP             SNP identifier
     
     TEST            Code indicating whether the sample is a case or control
     
     A1              Minor allele code
     
     A2              Major allele code
     
     GENO            Genotype counts: 11/12/22 
     
     O(HET)          Observed heterozygosity
     
     E(HET)          Expected heterozygosity
     
     P               P-value for Hardy-Weinberg test

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
head ./Data/QC/younameit.hwe
```

## An Example of Inclusion Thresholds for QC

Input files: in folder `./QC`

- 5SNPs.ped

- 5SNPs.map

The following command performs **multiple quality controls**. (The output will print all calculation progress generated by `PLINK`.)
1. **--mind** 0.1, exclude the individuals with missing rate larger than 0.1. You can change 0.1 to other desired values.
2. **--geno** 0.1, exclude the SNPs with missing rate larger than 0.1. You can change 0.1 to other desiered values.
3. **--maf** 0.05, exclude the SNPs with minor allele frequency (MAF) smaller than 0.05. You can change 0.05 to other desiered values.
4. **--hwe** 0.001, exclude the SNPs with P-values of Hardy-Weinberg test smaller than 0.001. You can change it to otehr desired values.
5. **--recode**, generate a new dataset.

```{bash eval=F}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/QC/5SNPs \
--mind 0.1 --geno 0.1 --maf 0.05 --hwe 0.001 --recode --out ./Data/QC/younameit
```
```{bash}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/QC/5SNPs \
--mind 0.1 --geno 0.1 --maf 0.05 --hwe 0.001 --recode --out ./Data/QC/younameit

```

The output files will be **"younameit.map"** and **"younameit.ped"** which have been quality controlled by the criteria you specified.

## Remove Closely Related Individuals
Input files: in folder `.QC`

- 5SNPs.ped

- 5SNPs.map

The following command will calculate **indentical by descent (IBS) distances** between all individuals. 
1. **--genome**, calculate IBS

```{bash eval=F}
# both Mac and Windows can run in Terminal
# modify the path to "plink" program

# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/QC/5SNPs --genome --out ./Data/QC/younameit
```
```{bash}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/QC/5SNPs --genome --out ./Data/QC/younameit
```

This will create the file **younameit.genome** which has the following fields:

     FID1      Family ID for first individual
     
     IID1      Individual ID for first individual
     
     FID2      Family ID for second individual
     
     IID2      Individual ID for second individual
     
     RT        Relationship type given PED file
     
     EZ        Expected IBD sharing given PED file
     
     Z0        P(IBD=0)
     
     Z1        P(IBD=1)
     
     Z2        P(IBD=2)
     
     PI_HAT    P(IBD=2)+0.5*P(IBD=1) ( proportion IBD )
     
     PHE       Pairwise phenotypic code (1,0,-1 = AA, AU and UU pairs)
     
     DST       IBS distance (IBS2 + 0.5*IBS1) / ( N SNP pairs )
     
     PPC       IBS binomial test
     
     RATIO     Of HETHET : IBS 0 SNPs (expected value is 2)

Scan the output file **"younameit.genome"** for any individuals with high **PI_HAT** values (e.g. greater than 0.05). 

Optionally, remove one member of the pair if close relatives are identified.

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
head ./Data/QC/younameit.genome
```
# An Imputation Example

In many cases, raw sequence data may have missing values. Then, an imputation step will be neccessary. **IMPUTE2** is a popular software package for imputation. Below is a simple example for conducting imputation by IMPUTE2. For more information about IMPUTE2, please refer its website http://mathgen.stats.ox.ac.uk/impute/impute_v2.html and the associated publication http://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1000529

Here is the example provided by **IMPUTE2 package**. The link is https://mathgen.stats.ox.ac.uk/impute/impute_v2.html#ex1

Files used for example are saved in folder `./Data/impute2_example`.

```{bash}
# bash codes- run in terminal
# IMPUTE2 only works for macOS or Linux users, Windows users can't run it

# macOS user only
# modify the path to "impute2" program
./Tools/IMPUTE2/impute_v2.3.2_MacOSX_Intel/impute2 \
-m ./Data/impute2_example/example.chr22.map \
-h ./Data/impute2_example/example.chr22.1kG.haps \
-l ./Data/impute2_example/example.chr22.1kG.legend \
-g ./Data/impute2_example/example.chr22.study.gens \
-strand_g ./Data/impute2_example/example.chr22.study.strand \
-int 20.4e6 20.5e6 -Ne 20000 \
-o ./Data/impute2_example/example.chr22.one.phased.impute2

```

# Population Stratification Correction Using EIGENSTRAT

The EIGENSTRAT method uses principal component analysis to explicitly model ancestry differences between cases and controls along continuous axes of variation; the resulting correction is specific to a candidate marker’s variation in frequency across ancestral populations, minimizing spurious associations while maximizing power to detect true associations. The EIGENSTRAT package has a built-in plotting script and supports multiple file formats and quantitative phenotypes. The package is based on the method described in [Price et al. 2006](https://www.ncbi.nlm.nih.gov/pubmed/16862161).

See https://github.com/DReichLab/EIG.

## EIGENSTRAT input files

**EIGENSTRAT** accepts multiple types of files as input, including "ANCESTRYMAP" files, "EIGENSTRAT" files, "PED" files, "PACKEDPED" files and "PACKEDANCESTRYMAP" files. Detailed introduction of these different file formats can be found in their [README](https://github.com/chrchang/eigensoft/blob/master/CONVERTF/README) file. We will use PED file format in this example.

We will prepare three files-- "genotype.ped", "genotype.pedsnp" and "genotype.pedind".

- The **PED files** format is described above. We will use genotype file: **genotype.ped** (saved in ./PC` folder and has size of 1.1G.)
- The **indiv file**(`./PC/genotype.pedind`) contains the first 6 or 7 columns of the genotype file.The genotype file is 1 line per individual.\
  The first 6 or 7 columns of the genotype file are:
    - 1st column is family ID.  
    - 2nd column is sample ID.
    - 3rd and 4th column are sample IDs of parents.  
    - 5th column is gender (male is 1, female is 2)
    - 6th column is case/control status (1 is control, 2 is case) OR
      quantitative trait value OR population group label.
    - 7th column (this column is optional) is always set to 1.  
  In the two genotype columns for each SNP, missing data is represented by 0.
- The **snp file**(`./PC/genotype.pedsnp`) contains 1 line per SNP.  There are 6 columns (last 2 optional):
  - 1st column is chromosome.  Use X for X chromosome.(Note: SNPs with illegal chromosome values, such as 0, will be removed)
  - 2nd column is SNP name
  - 3rd column is genetic position (in Morgans)
  - 4th column is physical position (in bases)
  - Optional 5th and 6th columns are reference and variant alleles.(For monomorphic SNPs, the variant allele can be encoded as X.)
  
The following command use the previous map file as `genotype.pedsnp` file (in Linux):

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
cp ./Data/PC/genotype_1k.map ./Data/PC/genotype_1k.pedsnp
```

The following command prepare the `genotype.pedind` file. Because the original data doesn't have case/control information, we will randomly assign them to case or control group to use as an example. 

```{r}
# R codes- run in R Console
set.seed(1)
group<-rep(1,344)
idx<-sample(344,170)
group[idx]<-2
write.table(group, "./Data/PC/group.txt", row.names=F, col.names=F, quote=F, sep=" ")
head(group)
```

Then we select the first five columns from `.ped` file (in Linux), add the group file we just generated to get the `genotype.pedind`. 

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
awk '{print $1,$2,$3,$4,$5}' ./Data/PC/genotype_1k.ped > ./Data/PC/genotype_ind.txt
paste -d' ' ./Data/PC/genotype_ind.txt ./Data/PC/group.txt> ./Data/PC/genotype_1k.pedind
```

## Run EIGENSTRAT Using Input Genotype Data

We call **smartpca.perl** in EIGENSTRAT on input genotype data. (All data and results in this section are saved in `.PC` folder.)

The options are:

**-i** example.ped : genotype file 

**-a** example.pedsnp : snp file 

**-b** example.pedind : indiv file 

**-k** : (Default is 10) number of principal components to output 

**-o** example.pca : output file of principal components

**-p** example.plot : prefix of output plot files of top 2 principal components. (labeling individuals according to labels in indiv file) 

**-e** example.eval : output file of all eigenvalues 

**-l** example.log : output logfile 

**The following code takes several minutes to run.** It will generate a warning message that `/usr/bin/gs: /apps/spack/bell/apps/libtiff/4.0.10-gcc-9.3.0-bjzustl/lib/libtiff.so.5: no version information available (required by /lib64/libgs.so.9)`. This is because the libtiff module called by this command is incompatible with the system gs command. You can safely ignore it.

```{bash eval=F}
# bash codes-can't run on either macOS or Windows
# this can only be run under Linux system or on computing server
./Tools/eigensoft-master/bin/smartpca.perl \
-i ./Data/PC/genotype_1k.ped -a ./Data/PC/genotype_1k.pedsnp -b ./Data/PC/genotype_1k.pedind \
-k 10 -o ./Data/PC/genotype_1k.pca -p ./Data/PC/genotype_1k.plot \
-e ./Data/PC/genotype_1k.eval -l ./Data/PC/genotype_1k.log
```

The main outputs are **"genotype.pca"**. A PCA plot is generated in **"genotype.plot.ps"**.

A **scree plot** displays the variance explained by each component. We look for the "elbow" of the plot, i.e., point where line bends. Could do formal test on derivative of scree line, but common sense approach often works fine.

We first extract eigenvalues from the PCA results file **"genotype.pca"** to save it in **"pc_eig.txt"**.

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
awk 'NR >= 2 && NR <= 11 {print}' ./Data/PC/genotype_1k.pca > ./Data/PC/pc_eig.txt
```

```{r}
# R codes- run in R Console
# generate plot in R
pc_eig <- read.table("./Data/PC/pc_eig.txt",header=F) # read in eigenvalue
names(pc_eig) <- "eig"

# generate scree plot-- eigenvalue vs # of PC
plot(1:10, pc_eig$eig, type="b", xlab="# PCs", ylab="Eigenvalues",
main="Scree plot")
```

Usually, we use a cut-off of eigenvalue > 1. In this plot, this criteria results in an unreasonable high number of factors, because the eigenvalue for PC 10 is still > 1.

## Test the Significance of Principle Components (PCs)

The following command creates a matrix from **genotype.pca** (in Linux)

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
tail -n+12 ./Data/PC/genotype_1k.pca > ./Data/PC/pc.txt
```

The following commands employ linear regression to the test the significance of PCs. P values can be found in the last column of the table: `Pr(>|t|)`.

- phenotype file: r_met.txt

```{r}
# R codes- run in R Console
y=read.table("./Data/PC/r_met.txt") #read phenotype data
pc=read.table("./Data/PC/pc.txt") #read pc data
y=as.matrix(y) #create a matrix from phenotype data
pc=as.matrix(pc) #create a matrix from pc data
fit=lm(y~pc)
summary(fit)

```

# Simple Association Test 

PLINK is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner, see http://zzz.bwh.harvard.edu/plink/.

The input files include PED file, MAP file, alternate phenotype file, and optional covariate file.

The PED file is a white-space (space or tab) delimited file, the first six columns are mandatory:

    Col 1: Family ID 

    Col 2: Individual ID 

    Col 3: Paternal ID 
    
    Col 4: Maternal ID 
    
    Col 5: Sex (1=male; 2=female; other character =unknown) 
    
    Col 6: Phenotype (The missing phenotype value for quantitative traits is, by default, -9)
    
    Col 7(*): Genotype (The missing genotype value is denoted as 0, by default)
    
Example: 

    FAM001 1 0 0 1 3.4 A A G G A C C C 

    FAM001 2 0 0 1 2.5 A A A G 0 0 A C
    
The MAP file has four columns:

    Col 1: Chromosome (1-22, X, Y or 0 if unplaced)
    
    Col 2: rs# or SNP identifier 
    
    Col 3: Genetic distance (morgans) 
    
    Col 4: Base-pair position (bp units)
    
To specify an alternate phenotype for analysis, i.e. other than the one in the .ped file, use the option --pheno. The alternate phenotype file has three columns: 

     Col 1: Family ID
     
     Col 2: Individual ID
     
     Col 3: Phenotype

Example: 

     F1     1110     2.30   22.22  2  
     
     F2     2202     34.12  18.23  1  
     
The phenotype can be either a quantitative trait or an affection status column. PLINK will automatically detect which type it is  based on whether a value other than 0, 1, 2 or the missing genotype code is observed.

(All files and results in this section are saved in `./PC` folder.)

## Quantitative Trait Association Test for Continuous Trait 

Below is an example that demonstrates how to use PLINK to perform a quantitative trait association test for a toy dataset with a continuous trait and 5 SNPs:

+ Genotype data:
  + 5SNPs.ped
  + 5SNPs.map
+ Phenotype data: 
    + conty.txt

The following commands perform the **simple association test**

1. **--assoc**, perform case/control or QTL association. You can also use **--fisher** to do Fisher's exact (allelic) test or **--model** to do Cochran-Armitage and full-model C/C association.
2. **--pheno**, specify the phenotype file.

```{bash eval=F}
# both Mac and Windows can run in Terminal

# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/Assoc/5SNPs --assoc --pheno ./Data/Assoc/conty.txt --out ./Data/Assoc/younameit.conty
```
```{bash}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/Assoc/5SNPs --assoc --pheno ./Data/Assoc/conty.txt --out ./Data/Assoc/younameit.conty
```

This will generate the output file **"younameit.conty.qassoc"** with following columns:

    CHR      Chromosome number

    SNP      SNP identifier

    BP       Physical position (base-pair)

    NMISS    Number of non-missing genotypes

    BETA     Regression coefficient

    SE       Standard error

    R2       Regression r-squared

    T        Wald test (based on t-distribution)

    P        Wald test asymptotic p-value
    
```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
head ./Data/Assoc/younameit.conty.qassoc
```

## Case/Control Asssociation Test for Binary Trait

Below is an example that demonstrates how to use PLINK to perform a case/control association test for a binary trait and 5 SNPs: 

- Genotype data:

  + 5SNPs.ped

  + 5SNPs.map
  

- Phenotype data: 

  + biny.txt
  
The following commands perform the **simple association test**

1. **--assoc**, perform case/control or QTL association. You can also use **--fisher** to do Fisher's exact (allelic) test or **--model** to do Cochran-Armitage and full-model C/C association.
2. **--pheno**, specify the phenotype file.

```{bash eval=F}
# both Mac and Windows can run in Terminal
# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/Assoc/5SNPs --assoc --pheno ./Data/Assoc/biny.txt --out ./Data/Assoc/younameit.biny
```
```{bash}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/Assoc/5SNPs --assoc --pheno ./Data/Assoc/biny.txt --out ./Data/Assoc/younameit.biny
```

The output is **"younameit.biny.assoc"**, which contains the fields:

     CHR     Chromosome
     SNP     SNP ID
     BP      Physical position (base-pair)
     A1      Minor allele name (based on whole sample)
     F_A     Frequency of this allele in cases
     F_U     Frequency of this allele in controls
     A2      Major allele name
     CHISQ   Basic allelic test chi-square (1df)
     P       Asymptotic p-value for this test
     OR      Estimated odds ratio (for A1, i.e. A2 is reference)
     
```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
head ./Data/Assoc/younameit.biny.assoc
```

## Quantitative Trait Association Test for Continuous Trait with Covariates

Below is an example that demonstrates how to use PLINK to perform a quantitative trait test for a continuous trait and 5 SNPs with 3 covariates:

1. Genotype data:
  + 5SNPs.ped
  + 5SNPs.map
2. Phenotype data:
  + conty_pheno.txt
3. Covariates:
  + conty_covar.txt

Before the association test, we adjust the phenotypes to remove the effects of covariates. For the continous phenotype, we employ linear regression to adjust the phenotype. Then, we use the residual as the new phenotype in the association test.

### Covariates adjustments (using R):

```{r}
## R codes- run in R Console
# Read phenotype data and covariates data
pheno=read.table("./Data/Assoc/conty_pheno.txt") #read phenotype data
covar=read.table("./Data/Assoc/conty_covar.txt") #read covariates data
pheno=as.matrix(pheno) #create a matrix from the phenotype data 
covar=as.matrix(covar) #create a matrix from the covariates data
n=dim(pheno)[1] #sample size
p=dim(pheno)[2] #number of traits
```

Covariates adjustment is done by fitting a linear regression model and obtain residuals as new phenotype data for association test.

The following codes construct the linear model for each of the traits using `lm` function.The resulted residual data is saved for analysis.

```{r}
# R codes- run in R Console
fit=list() #create an empty list
residpheno=matrix(0, n, p) #create an empty matrix
for (i in 1:p){
  fit[[i]]=lm(pheno[,i]~covar) #for each trait, fit a linear regression model of phenotype against covariates
  residpheno[,i]=resid(fit[[i]]) #obtain residual phenotype data
}
write.table(residpheno, "./Data/Assoc/resid_phenotype.txt", row.names=F, col.names=F, quote=F, sep=" ") #save residual phenotype data
```

### Residual phenotype data after covariates adjustments (in UNIX):

This code combines the above residual data and sample ID information to contruct the phenotype data for PLINK.

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
paste ./Data/Assoc/famID.txt ./Data/Assoc/sampleID.txt \
./Data/Assoc/resid_phenotype.txt > ./Data/Assoc/resid_phenotype_assoc.txt
```

Then we can use PLINK to perform the association test.

### Association test using plink

After adjusting the phenotype using a set of covariates, we use the residual as new phenotype for the association test. `PLINK` was used again to perform assiciation test. 

```{bash eval=F}
# both Mac and Windows can run in Terminal
# Windows user
./Tools/PLINK/plink_win64/plink.exe \
--noweb --file ./Data/Assoc/5SNPs --assoc --pheno ./Data/Assoc/resid_phenotype_assoc.txt \
--out ./Data/Assoc/younameit_resid
```
```{bash}
# macOS user
./Tools/PLINK/plink_mac/plink \
--noweb --file ./Data/Assoc/5SNPs --assoc --pheno ./Data/Assoc/resid_phenotype_assoc.txt \
--out ./Data/Assoc/younameit_resid
```

```{bash}
# bash codes- run in terminal
# Windows user need to run in Powershell App
### check association results
head ./Data/Assoc/younameit_resid.qassoc
```

This output file **"younameit.qassoc"** has the following columns:

    CHR      Chromosome number

    SNP      SNP identifier

    BP       Physical position (base-pair)

    NMISS    Number of non-missing genotypes

    BETA     Regression coefficient

    SE       Standard error

    R2       Regression r-squared

    T        Wald test (based on t-distribution)

    P        Wald test asymptotic p-value
    
In the next section, we will start analyzing real data from a case study. The procedures might not be exactly the same here, as real data require more proprocessing steps before it can be used for association study.




<!--chapter:end:02-gwastool.Rmd-->


# GWAS Case Study-Data Preparation

Placeholder


## Environment setting and packages
## Accessing data
### Data availability
### Data files documentation
## Preprocessing Phenotype Data
### Extracting phenotype data
### Summary of Phenotypes Data
### Histogram of each trait
### Handling the missing values in the phenotypes and covariates
### Checking normality assumptions for each trait
### Normalize phenotype data
## Covariates Adjustment
## Prepare PLINK data
### Step 1: prepare PLINK ped file
## Step 1: prepare PLINK ped file
### Step 2: prepare PLINK map file
### Step 3 : prepare PLINK alternate phenotype file (in Linux)

<!--chapter:end:03-gwas_part1.Rmd-->


# GWAS with Continuous Trait

Placeholder


## Manhattan Plot and Q-Q Plot
### Generate manhattan plot
### Generate Q-Q plot
## Multiple-Test Correction
### Prepare R input files (in Linux)
### Significant SNPs after a multiple-test correction of FDR (using R)
## Regional Association Plot
### Obtain regional SNPs (using R)
## Distribution of Z-endoxifen ratio for rs8138080
### Obtain phenotype data (Z-endoxifen ratio) (using R)
### Obtain genotype for significant SNPs (using R)
### Create Box plot (using R)
## Proportion of Variation Explained by rs8138080
### Recode genotype of significant SNPs (using R)
### Calculate marginal $R^2$ (using R)
## PRACTICE TIME
### Practice 1
### Practice 2

<!--chapter:end:04-gwas_part2.Rmd-->


# GWAS with Binary Trait

Placeholder


## Data Preparation
## Genome-Wide Case/Control Association Test (in PLINK)
## Manhattan Plot and Q-Q Plot
## Generate Manhattan plot
## Generate Q-Q plot
## Multiple-Test Correction
## Top 10 SNPs

<!--chapter:end:05-gwas_part3.Rmd-->


# GxG and GxE Interactions, Power and Sample Size Calculation

Placeholder


## Prepare Files
## Test for Epistasis (GxG)
## Gene-Environment Interaction (GxE)
## Power and sample size calculation for GWAS
### Power calculation
#### Case study with binary outcome:
#### Case study with continuous outcome:
#### Continuous study with continuous GxE interaction:
### Sample size calculation
#### Case study with binary outcome:
#### Case study with continuous outcome
#### Binary study with binary GxE interactions

<!--chapter:end:06-gwas_gxe.Rmd-->


# LDSC and Gene-Set Analysis

Placeholder


## LDSC (Linkage Disequilibrium Score Regression)
## MAGMA (for Gene-Level Analysis)
### Prepare Annotation File
### Use MAGMA for Gene-Set Analysis

<!--chapter:end:07-post_GWAS.Rmd-->


# Perform eQTL Analysis

Placeholder


## Filtering SNPs by MAF
## Gene expression profiling
## Covariates
## Using principle components as covariates
## eQTL analysis
### cis-eQTL and trans-eQTL analysis
## Multiple testing correction
## Gene regional Manhanttan plots 

<!--chapter:end:08-eqtl.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}
'`

<!--chapter:end:09-reference.Rmd-->

