--- 
title: "2024 CSHL course - GWAS Hands-on Analysis"
author: "Danni Liu - Purdue University"
date: "`r Sys.Date()`"
site: bookdown::bookdown_site
output: bookdown::gitbook
documentclass: book
bibliography: [book.bib, packages.bib]
biblio-style: apalike
link-citations: yes
github-repo: rstudio/cshl_2024
description: "This is the course material for the 2024 CSHL hands-on session in GWAS."
---

# Welcome to the GWAS and eQTL Workshop

Placeholder


### Overview
### Folder structure:
### Programming Languages: 
### Required GWAS Tools

<!--chapter:end:index.Rmd-->


# An Overview of Regression with Toy Examples

Placeholder


## Environment Setting and Packages
## Toy Example Data
## Logistic Regression
## ROC curve and AUC for HOXB13
## Practice
## Comparison of HOXB13 ROC curve and NHLH1 ROC curve
## Linear Regression
### Linear regression of HOXB13 vs. IL17BR

<!--chapter:end:01-intro.Rmd-->


# GWAS Tools and Data Preprocessing Workflow with Toy Example 

Placeholder


## Environment Setting
## GWAS Tools
## Quality Control
### An Example of Summary Statistics
#### Input files
#### Calculating missing rate
#### Calculating allel frequencies
#### Hardy-Weinberg Equilibrium
### An Example of Inclusion Thresholds for QC
### Remove Closely Related Individuals
## An Imputation Example
## Population Stratification Correction Using PLINK
### PCA on whole genome data using PLINK
### Test the Significance of Principle Components (PCs)
## Simple Association Test 
### Quantitative Trait Association Test for Continuous Trait 
### Case/Control Asssociation Test for Binary Trait
## Quantitative Trait Association Test for Continuous Trait with Covariates
### Covariates adjustments (using R):
### Association test using plink

<!--chapter:end:02-gwastool.Rmd-->


# GWAS Case Study-Data Preparation

Placeholder


## Environment setting and packages
## Accessing data
### Data availability
### Data files documentation
## Preprocessing Phenotype Data
### Extracting phenotype data
### Summary of Phenotypes Data
### Histogram of each trait
### Handling the missing values in the phenotypes and covariates
### Checking normality assumptions for each trait
### Normalize phenotype data
## Covariates Adjustment
## Prepare PLINK data
### Step 1: prepare PLINK ped file
## Step 1: prepare PLINK ped file
### Step 2: prepare PLINK map file
### Step 3 : prepare PLINK alternate phenotype file (in Linux)

<!--chapter:end:03-gwas_part1.Rmd-->


# GWAS with Continuous Trait

Placeholder


## Manhattan Plot and Q-Q Plot
### Generate manhattan plot
### Generate Q-Q plot
## Multiple-Test Correction
### Prepare R input files (in Linux)
### Significant SNPs after a multiple-test correction of FDR (using R)
## Regional Association Plot
### Obtain regional SNPs (using R)
## Distribution of Z-endoxifen ratio for rs8138080
### Obtain phenotype data (Z-endoxifen ratio) (using R)
### Obtain genotype for significant SNPs (using R)
### Create Box plot (using R)
## Proportion of Variation Explained by rs8138080
### Recode genotype of significant SNPs (using R)
### Calculate marginal $R^2$ (using R)
## PRACTICE TIME
### Practice 1
### Practice 2

<!--chapter:end:04-gwas_part2.Rmd-->


# GWAS with Binary Trait

Placeholder


## Data Preparation
## Genome-Wide Case/Control Association Test (in PLINK)
## Manhattan Plot and Q-Q Plot
## Generate Manhattan plot
## Generate Q-Q plot
## Multiple-Test Correction
## Top 10 SNPs

<!--chapter:end:05-gwas_part3.Rmd-->


# GxG and GxE Interactions, Power and Sample Size Calculation

Placeholder


## Prepare Files
## Test for Epistasis (GxG)
## Gene-Environment Interaction (GxE)
## Power and sample size calculation for GWAS
### Power calculation
#### Case study with binary outcome:
#### Case study with continuous outcome:
#### Continuous study with continuous GxE interaction:
### Sample size calculation
#### Case study with binary outcome:
#### Case study with continuous outcome
#### Binary study with binary GxE interactions

<!--chapter:end:06-gwas_gxe.Rmd-->


# LDSC and Gene-Set Analysis

Placeholder


## LDSC (Linkage Disequilibrium Score Regression)
## MAGMA (for Gene-Level Analysis)
### Prepare Annotation File
### Use MAGMA for Gene-Set Analysis
### Compare the Manhattan plots

<!--chapter:end:07-post_GWAS.Rmd-->

# Perform eQTL Analysis

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
knitr::opts_knit$set(root.dir = "./Data/eqtl")
getwd
```

We will use "MatrixEQTL" package in R. To install the package use the codes: `install.packages("MatrixEQTL")`

```{r eval=T, message=FALSE, warnings=FALSE}
# R codes- run in R Console
#load R packages
library(data.table)
library(readr)
library(tidyr)
library(dplyr)
library(MASS)
library(MatrixEQTL)
```


```{bash eval=F}
# change the path below to where you saved "cshl_2024" folder
## run in R Terminal
#mydir="/path/to/cshl_2024/Data/eqtl"
mydir="./Data/eqtl"
cd $mydir
```

```{r eval=F}
# R codes- run in R Console
#setwd("/path/to/cshl_2024/Data/eqtl")
setwd("./Data/eqtl")
mydir<-getwd()
```

Files: 
- `SNP.txt`: rows are SNPs with row names as SNP IDs, and columns are sample IDs.
- `gexp.txt`: rows are genes are and columns are sample IDs.


```{r}
snps <- read.table("geno.txt",header=T,row.names = 1)
expr <-read.table("gexp.txt",header=T,row.names = 1)
```

```{r}
dim(expr)
head(expr)[1:12]
```

```{r}
dim(snps)
head(snps)[1:12]
```


```{r}
maf<-read.table("geno_maf.txt")$V1

truehist(maf, main = "Histogram of MAF values.", col = "steelblue")
lines(density(maf), lty = 2, col = "darkorange", lwd = 3)
```

## Filtering SNPs by MAF
In an eQTL study often a minimum MAF is required. Since MAF essentially reflects how often an allele has been observed in a population, it also defines how often the gene expression levels have been observed for heterozygous and homozygous alleles.

Calculate the number of heterozygous and homozygous observations expected for SNPs with a MAF of 5%, 10% and 15% given Hardy-Weinberg equilibrium. What are useful MAF thresholds for SNPs to include in an eQTL analysis?

```{r}
p = c(0.05, 0.1, 0.15)
q = 1-p
# Calulate frequency of minor allele being present in homozygous and heterozygous state
f_hom = p^2
f_het = 2*p*q
# Expected number of observations in a sample size of 10000
sample_size = 500
round(f_hom * sample_size)

round(f_het * sample_size)
```

## Gene expression profiling
Now that we have an idea of what is stored in the genotype data frame let’s take a look at the expression data. For eQTL analyses it is important for the gene expression to be normally distributed among samples, therefore RNA-seq data has to be transformed by, for example quantile normalization.

Let’s check the distribution of gene expression levels across samples for the first gene in our expr data frame.

```{r}
gname = rownames(expr)[1]
truehist(as.numeric(filter(expr, rownames(expr) == gname)[-1]), main = paste("Gene expression profile for gene:",gname), xlab = "Expression level", col = "darkorange")
lines(density(as.numeric(filter(expr, rownames(expr) == gname)[-1])), lty = 2, col = "steelblue", lwd = 3)
```

Now we’ll plot the expression levels of the first gene against the first SNP, 10th gene against the 10th SNP, 10th gene against the 1th SNP, as well as the 10th gene against the 6th SNP, depending on the genotypes of the samples by using simple dot plots. We’ll add a bit of random noise (jitter) to the genotype data to make it all look more comprehensible.

```{r}
par(mfrow=c(1,4))

plot(jitter(as.numeric(snps[1,]), factor = 0.5), as.numeric(expr[1,]),
     xlab = rownames(snps)[1], ylab = rownames(expr)[1], col = "steelblue",
     main = paste(rownames(expr)[1], "vs", rownames(snps)[1]), xlim= c(-0.5,2.5), xaxt="n")
axis(1, at =c (0,1,2), labels = c("0", "1", "2"))
plot(jitter(as.numeric(snps[10,]), factor = 0.5), as.numeric(expr[10,]),
     xlab = rownames(snps)[10], ylab = rownames(expr)[10], col = "steelblue",
     main = paste(rownames(expr)[10], "vs", rownames(snps)[10]), xlim= c(-0.5,2.5), xaxt="n")
axis(1, at =c (0,1,2), labels = c("0", "1", "2"))
plot(jitter(as.numeric(snps[1,]), factor = 0.5), as.numeric(expr[10,]),
     xlab = rownames(snps)[1], ylab = rownames(expr)[10], col = "steelblue",
     main = paste(rownames(expr)[10], "vs", rownames(snps)[1]), xlim= c(-0.5,2.5), xaxt="n")
axis(1, at =c (0,1,2), labels = c("0", "1", "2"))
plot(jitter(as.numeric(snps[6,]), factor = 0.5), as.numeric(expr[10,]),
     xlab = rownames(snps)[6], ylab = rownames(expr)[10], col = "steelblue",
     main = paste(rownames(expr)[10], "vs", rownames(snps)[6]), xlim= c(-0.5,2.5), xaxt="n")
axis(1, at =c (0,1,2), labels = c("0", "1", "2"))
```

## Covariates

We can also add covariates to the analysis.
```{r}
covar<-read.table("covariates.txt",header=T,sep="\t")
dim(covar)
head(covar)
```

But due to the requirement of the `MatrixEQTL` package, categorical covariates need to be recoded as 0,1,2...
```{r}
covar_recode<-covar                                                                  
covar_recode$gender[covar_recode$gender=="male"]<-1
covar_recode$gender[covar_recode$gender=="female"]<-0
covar_recode$race[covar_recode$race=="white"]<-1
covar_recode$race[covar_recode$race=="black or african american"]<-0
write.table(t(covar_recode),"covar_recode.txt",quote=F, sep=" ",row.names=T,col.names=F)
```


## Using principle components as covariates
Further we will explore the use of principle components of genotype as covariates in linear models of gene expression to account for unknown sources of variation from genotype. 

```{r}
pca <- prcomp(t(snps), center=TRUE, scale = TRUE)
sum(pca$sdev[1:5]^2)/sum(pca$sdev^2) # percentage of variance explained by first 5 PCs
screeplot(pca,type="lines") 
```

## eQTL analysis
Now that we’ve covered the basics let’s focus on real world expression and genotyping data. We’ll be using R’s `MatrixEQTL` package which is designed for fast eQTL analysis on large genomic data sets. `MatrixEQTL` can test for association between genotype and gene expression using linear regression with either additive or ANOVA genotype effects. The models can include covariates to account for factors as population stratification, gender, and clinical variables. It also supports models with heteroscedastic and/or correlated errors, false discovery rate estimation and separate treatment of local (cis) and distant (trans) eQTLs. You can learn more about MatrixEQTL [here](http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/).

### cis-eQTL and trans-eQTL analysis
To perform an eQTL analysis we not only need to know the genotype and gene expression values for every sample, but also the genomic positions of genes and SNPs. This is necessary to define which SNPs should be tested against which genes. For cis-eQTL analyses, SNPs in proximity to the gene are chosen and for trans-eQTL analyses SNPs further away, or on different chromosomes, are taken into account. The window in cis-eQTL analysis is commonly chosen to be 500kb-1Mb measured from gene’s TSS.

In addition, the main `MatrixEQTL` function `Matrix_eQTL_main` requires several extra parameters for cis/trans analysis:

- `pvOutputThreshold.cis` : p-value threshold for cis-eQTLs.
- `output_file_name.cis` : detected cis-eQTLs are saved in this file.
- `cisDist` – maximum distance at which gene-SNP pair is considered local.
- `snpspos` : data frame with information about SNP locations, must have 3 columns - SNP name, chromosome, and position. See sample SNP location file.
- `genepos` : data frame with information about gene locations, must have 4 columns - the name, chromosome, and positions of the left and right ends. See sample gene location file.

We first setup the input files and the above parameters for eQTL analysis:
```{r}
# eQTL mapping, cis/trans, no pcs
suppressMessages(library(MatrixEQTL))

SNP_file_name <-  "geno.txt";
snps_location_file_name <- "snppos.txt";
expression_file_name <- "gexp.txt";
gene_location_file_name <- "genepos.txt";
covariates_file_name <- "covar_recode.txt";
cis_threshold <- 0.01
trans_threshold <- 0.01
cis_dist <- 1e6
  
# Output file name
output_file_name_cis = tempfile();
output_file_name_tra = tempfile();
  
## Settings
# Linear model to use, modelANOVA, modelLINEAR, or modelLINEAR_CROSS
useModel = modelLINEAR; 

# Only associations significant at this level will be saved
pvOutputThreshold_cis = cis_threshold; 
pvOutputThreshold_tra = trans_threshold;
  
# Set to character() for no covariates
# covariates_file_name = character();
  
# Error covariance matrix
# Set to numeric() for identity.
errorCovariance = numeric();
  
# Distance for local gene-SNP pairs
cisDist = cis_dist
```

The following codes performs both cis- and tran-eQTL analysis using the above setup files and parameters. The analysis is run by calling `Matrix_eQTL_main`, but the input data files need to be stored in required data object structure before it can be called to perform the analysis. The following codes first tell R how to read in the files correctly by specifying delimeters, etc. Then, both the input files and setting parameters were fed to the `Matrix_eQTL_main` options.

```{r}
## Load genotype data
snps = SlicedData$new();
snps$fileDelimiter = " "; # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1; # one row of column labels
snps$fileSkipColumns = 1; # one column of row labels
snps$fileSliceSize = 2000; # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);
  
## Load gene expression data
gene = SlicedData$new();
gene$fileDelimiter = " "; # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1; # one row of column labels
gene$fileSkipColumns = 1; # one column of row labels
gene$fileSliceSize = 2000; # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);
  
## Load covariates
cvrt = SlicedData$new();
cvrt$fileDelimiter = " "; # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1; # one row of column labels
cvrt$fileSkipColumns = 1; # one column of row labels
if(length(covariates_file_name)>0) {
  cvrt$LoadFile(covariates_file_name);
}
  
## Run the analysis
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE);
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE);
  
me = Matrix_eQTL_main(
  snps = snps,
  gene = gene,
  cvrt = cvrt,
  output_file_name = output_file_name_tra,
  pvOutputThreshold = pvOutputThreshold_tra,
  useModel = useModel,
  errorCovariance = errorCovariance,
  verbose = FALSE,
  output_file_name.cis = output_file_name_cis,
  pvOutputThreshold.cis = pvOutputThreshold_cis,
  snpspos = snpspos,
  genepos = genepos,
  cisDist = cisDist,
  pvalue.hist = TRUE,
  min.pv.by.genesnp = FALSE,
  noFDRsaveMemory = FALSE);

## for generating QQ plot  
me_qq = Matrix_eQTL_main(
  snps = snps,
  gene = gene,
  cvrt = cvrt,
  output_file_name = output_file_name_tra,
  pvOutputThreshold = pvOutputThreshold_tra,
  useModel = useModel,
  errorCovariance = errorCovariance,
  verbose = FALSE,
  output_file_name.cis = output_file_name_cis,
  pvOutputThreshold.cis = pvOutputThreshold_cis,
  snpspos = snpspos,
  genepos = genepos,
  cisDist = cisDist,
  pvalue.hist = "qqplot",
  min.pv.by.genesnp = FALSE,
  noFDRsaveMemory = FALSE)
  
unlink(output_file_name_tra);
unlink(output_file_name_cis);
  
cat('Analysis done in: ', me$time.in.sec, ' seconds', '\n');
```

Check the results from both cis- and tran-eQTL analysis.

```{r}
head(me$cis$eqtls)
```

```{r}
head(me$trans$eqtls)
```

```{r eval=T,echo=F, fig.cap='Histogram of p-values for cis- and trans-eQTL', out.width='100%', fig.asp=1.2, fig.align='center'}
## Make the histogram of local and distant p-values
# y-axis is density, which is rescaled frequency to make the sum of area of bars equal to 1
plot(me)

# to look at original frequency, use this code
# hist(me$cis$eqtls$pvalue)
# hist(me$trans$eqtls$pvalue)
```

```{r}
## Make the qq-plot of local and distant p-values
plot(me_qq)  
```

## Multiple testing correction
Whenever multiple statistical tests are performed, a multiple testing correction has to be performed. This is necessary because many hypotheses are tested. Therefore each calculated association p-value has to be corrected for multiple testing. `MatrixEQTL` does this for you automatically and returns the corrected p-value as a false discovery rate (`FDR`). Common thresholds on FDR are 5% or 10%.

## Gene regional Manhanttan plots 
Manhattan plots are a way to depict association p-values of multiple SNPs at once. They are also very common in GWAS. Manhattan plots are a rather convinient modality for interpretation of eQTL signals in terms of LD.

Let’s now generate a manhattan plot for gene ENSG00000172404.4, plotting the base-pair position on the x-axis and the −log10(pvalue)
 of the SNP in the y axis. Manhattan plots usually depict all tested SNPs, not only the ones passing a certain p-value threshold. Therefore we’ll first obtain all the association p-values for all tested SNPs for gene ENSG00000172404.4.
 
```{r}
gene_id = "CHD5"

gene_values = read.table(expression_file_name, row.names=1, header=TRUE)
single_gene_exp = SlicedData$new()
single_gene_exp$CreateFromMatrix(as.matrix(gene_values[gene_id, , drop=FALSE]))
single_gene_exp
```

```{r}
snpspos = read.table(snps_location_file_name, 
                     header = TRUE, 
                     stringsAsFactors = FALSE)

genepos = read.table(gene_location_file_name, 
                     header = TRUE, 
                     stringsAsFactors = FALSE)

single_cis_eqtl_res = Matrix_eQTL_main(snps, 
                                       single_gene_exp,
                                       verbose = FALSE,
                                       output_file_name.cis = NULL,
                                       output_file_name = NULL,
                                       pvOutputThreshold.cis=1, 
                                       snpspos = snpspos,
                                       genepos = genepos)

manh_data = merge(single_cis_eqtl_res$cis$eqtls, snpspos, by.x = "snps", by.y = "snpid")
manh_data = manh_data [,c("pos", "chr", "pvalue", "snps")]
head(manh_data)
```

```{r}
# Plot the Manhattanplot
with(manh_data ,plot(pos, -log10(pvalue), xlab = "genomic position (bp)",
main=paste(gene_id, "associated SNPs")))
# Highlight the lead SNP
with(manh_data[which.min(manh_data$pvalue),,drop=FALSE] ,
points(pos, -log10(pvalue), pch=20, col="red"))
# Add a label to the lead SNP
with(manh_data[which.min(manh_data$pvalue),,drop=FALSE],
text(pos + diff(range(manh_data$pos))*0.2, -log10(pvalue), labels = snps))
```

<!--chapter:end:08-eqtl.Rmd-->

`r if (knitr:::is_html_output()) '
# References {-}
'`

<!--chapter:end:09-reference.Rmd-->

