<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 An Overview of Regression with Toy Examples | 2024 CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 An Overview of Regression with Toy Examples | 2024 CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  <meta name="github-repo" content="rstudio/cshl_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 An Overview of Regression with Toy Examples | 2024 CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  

<meta name="author" content="Danni Liu - Purdue University" />


<meta name="date" content="2024-06-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.4</b> An Example of Summary Statistics</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#input-files"><i class="fa fa-check"></i><b>3.4.1</b> Input files</a></li>
<li class="chapter" data-level="3.4.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-missing-rate"><i class="fa fa-check"></i><b>3.4.2</b> Calculating missing rate</a></li>
<li class="chapter" data-level="3.4.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-allel-frequencies"><i class="fa fa-check"></i><b>3.4.3</b> Calculating allel frequencies</a></li>
<li class="chapter" data-level="3.4.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>3.4.4</b> Hardy-Weinberg Equilibrium</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.5</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.6</b> Remove Closely Related Individuals</a></li>
<li class="chapter" data-level="3.7" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#eigenstrat-input-files"><i class="fa fa-check"></i><b>3.7</b> EIGENSTRAT input files</a></li>
<li class="chapter" data-level="3.8" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#run-eigenstrat-using-input-genotype-data"><i class="fa fa-check"></i><b>3.8</b> Run EIGENSTRAT Using Input Genotype Data</a></li>
<li class="chapter" data-level="3.9" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>3.9</b> Test the Significance of Principle Components (PCs)</a></li>
<li class="chapter" data-level="3.10" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>3.10</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="3.11" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>3.11</b> Case/Control Asssociation Test for Binary Trait</a></li>
<li class="chapter" data-level="3.12" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>3.12</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="3.12.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>3.12.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="3.12.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#residual-phenotype-data-after-covariates-adjustments-in-unix"><i class="fa fa-check"></i><b>3.12.2</b> Residual phenotype data after covariates adjustments (in UNIX):</a></li>
<li class="chapter" data-level="3.12.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#association-test-using-plink"><i class="fa fa-check"></i><b>3.12.3</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html"><i class="fa fa-check"></i><b>4</b> Unit 3. GWAS Case Study-Continuous Trait</a>
<ul>
<li class="chapter" data-level="4.1" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html#data-preprocessing"><i class="fa fa-check"></i><b>4.1</b> Data Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>5</b> Applications</a>
<ul>
<li class="chapter" data-level="5.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>5.1</b> Example one</a></li>
<li class="chapter" data-level="5.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>5.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>6</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">2024 CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="an-overview-of-regression-with-toy-examples" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> An Overview of Regression with Toy Examples<a href="an-overview-of-regression-with-toy-examples.html#an-overview-of-regression-with-toy-examples" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p><span style="color: #0064a4;"><br />
<strong>Note:</strong> In order to run codes in this notebook, please make sure you have downloaded the “bigcare-main” folder from GitHub and have RStudio installed on your laptop.
</span></p>
</blockquote>
<p>This notebook demonstrates two basic statistical regression models using toy examples.</p>
<ul>
<li>case/control study by logistic regression</li>
<li>quantitative trait study by linear regression</li>
</ul>
<p>We use only R codes in this notebook.</p>
<div id="environment-setting-and-packages" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Environment Setting and Packages<a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before we can run any analysis, we need to set the working directory path in RStudio Console to the downloaded “bigcare-main” folder and load the necessary R packages. To setup the working directory, make sure you run the following codes in R Terminal.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="an-overview-of-regression-with-toy-examples.html#cb1-1" tabindex="-1"></a><span class="co"># change the path below to where you saved &quot;bigcare-main&quot; folder</span></span>
<span id="cb1-2"><a href="an-overview-of-regression-with-toy-examples.html#cb1-2" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/path/to/gwas&quot;</span>)</span></code></pre></div>
<p>In this notebook, we will use R packages “data.table”, “GEOquery”, “plyr” and “ROCR”. You can install the R base packages using install.packages(). “GEOquery” is a BioConductor package that needs to be installed with BioManager. The codes below show you how to install these packages in R. They are commented out with # here, so if you want to run them on your own laptop, please remove # before the codes.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="an-overview-of-regression-with-toy-examples.html#cb2-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;data.table&quot;</span>)</span>
<span id="cb2-2"><a href="an-overview-of-regression-with-toy-examples.html#cb2-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;plyr&quot;</span>)</span>
<span id="cb2-3"><a href="an-overview-of-regression-with-toy-examples.html#cb2-3" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ROCR&quot;</span>)</span>
<span id="cb2-4"><a href="an-overview-of-regression-with-toy-examples.html#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="an-overview-of-regression-with-toy-examples.html#cb2-5" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb2-6"><a href="an-overview-of-regression-with-toy-examples.html#cb2-6" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb2-7"><a href="an-overview-of-regression-with-toy-examples.html#cb2-7" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;GEOquery&quot;</span>)</span></code></pre></div>
<p>After installation, The list of packages are loaded with library() function.</p>
<p>The following code might generate multiple messages when loading the packages, such as “The following objects are masked from XXX”, which is normal. You can ignore them.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="an-overview-of-regression-with-toy-examples.html#cb3-1" tabindex="-1"></a><span class="co">#load R packages</span></span>
<span id="cb3-2"><a href="an-overview-of-regression-with-toy-examples.html#cb3-2" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb3-3"><a href="an-overview-of-regression-with-toy-examples.html#cb3-3" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb3-4"><a href="an-overview-of-regression-with-toy-examples.html#cb3-4" tabindex="-1"></a><span class="fu">library</span>(plyr)</span>
<span id="cb3-5"><a href="an-overview-of-regression-with-toy-examples.html#cb3-5" tabindex="-1"></a><span class="fu">library</span>(ROCR)</span>
<span id="cb3-6"><a href="an-overview-of-regression-with-toy-examples.html#cb3-6" tabindex="-1"></a><span class="fu">library</span>(knitr)</span></code></pre></div>
</div>
<div id="toy-example-data" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> Toy Example Data<a href="an-overview-of-regression-with-toy-examples.html#toy-example-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To illustrate the statistical methods, we will use a small dataset of 60 samples with both continuous and binary variables. This dataset is a microarray expression data downloaded from GSE1378. It contains 60 patients with ER-positive primary breast cancer treated with tamoxifen monotherapy for 5 years. We only extracted the recurrent status and gene expression of 3 genes to this toy example data. You can load this extracted data as below with fread function:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="an-overview-of-regression-with-toy-examples.html#cb4-1" tabindex="-1"></a>toy_data <span class="ot">&lt;-</span> <span class="fu">fread</span>(<span class="st">&quot;./data/toy_example_data.txt&quot;</span>)</span></code></pre></div>
<p>We could take a view at the data. head function prints the first 6 rows of the dataset. summary function provides summary statistics of the numeric variables. Column status is a binary variable indicating whether the sample is recurrence or non-recurrence. The last three columns are expression of three genes HOXB13, IL17BR and NHLH1, so they are continuous.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="an-overview-of-regression-with-toy-examples.html#cb5-1" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(toy_data))</span></code></pre></div>
<table>
<colgroup>
<col width="9%" />
<col width="18%" />
<col width="5%" />
<col width="27%" />
<col width="13%" />
<col width="13%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">sampleID</th>
<th align="left">status</th>
<th align="right">time</th>
<th align="left">ratio_group</th>
<th align="right">gene_HOXB13</th>
<th align="right">gene_IL17BR</th>
<th align="right">gene_NHLH1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GSM22365</td>
<td align="left">Status=recur</td>
<td align="right">75</td>
<td align="left">High HOXB13:IL17BR ratio</td>
<td align="right">-2.8968509</td>
<td align="right">-2.4744445</td>
<td align="right">-0.1214872</td>
</tr>
<tr class="even">
<td align="left">GSM22366</td>
<td align="left">Status=recur</td>
<td align="right">21</td>
<td align="left">High HOXB13:IL17BR ratio</td>
<td align="right">-2.9811459</td>
<td align="right">-2.1587305</td>
<td align="right">-0.6215178</td>
</tr>
<tr class="odd">
<td align="left">GSM22367</td>
<td align="left">Status=recur</td>
<td align="right">38</td>
<td align="left">Low HOXB13:IL17BR ratio</td>
<td align="right">-1.3201379</td>
<td align="right">1.0535341</td>
<td align="right">0.0583918</td>
</tr>
<tr class="even">
<td align="left">GSM22368</td>
<td align="left">Status=recur</td>
<td align="right">43</td>
<td align="left">High HOXB13:IL17BR ratio</td>
<td align="right">0.5141317</td>
<td align="right">-1.7643865</td>
<td align="right">0.0073468</td>
</tr>
<tr class="odd">
<td align="left">GSM22369</td>
<td align="left">Status=non-recur</td>
<td align="right">169</td>
<td align="left">Low HOXB13:IL17BR ratio</td>
<td align="right">-3.0522629</td>
<td align="right">0.8341051</td>
<td align="right">-0.5538062</td>
</tr>
<tr class="even">
<td align="left">GSM22370</td>
<td align="left">Status=recur</td>
<td align="right">42</td>
<td align="left">Low HOXB13:IL17BR ratio</td>
<td align="right">-3.2735509</td>
<td align="right">-0.9047099</td>
<td align="right">0.8428828</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="an-overview-of-regression-with-toy-examples.html#cb6-1" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">summary</span>(toy_data))</span></code></pre></div>
<table>
<colgroup>
<col width="2%" />
<col width="14%" />
<col width="14%" />
<col width="12%" />
<col width="14%" />
<col width="13%" />
<col width="13%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">sampleID</th>
<th align="left">status</th>
<th align="left">time</th>
<th align="left">ratio_group</th>
<th align="left">gene_HOXB13</th>
<th align="left">gene_IL17BR</th>
<th align="left">gene_NHLH1</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"></td>
<td align="left">Length:60</td>
<td align="left">Length:60</td>
<td align="left">Min. : 5.00</td>
<td align="left">Length:60</td>
<td align="left">Min. :-3.8524</td>
<td align="left">Min. :-4.1248</td>
<td align="left">Min. :-1.27238</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">Class :character</td>
<td align="left">Class :character</td>
<td align="left">1st Qu.: 46.25</td>
<td align="left">Class :character</td>
<td align="left">1st Qu.:-3.3163</td>
<td align="left">1st Qu.:-1.4710</td>
<td align="left">1st Qu.:-0.48316</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">Mode :character</td>
<td align="left">Mode :character</td>
<td align="left">Median : 88.00</td>
<td align="left">Mode :character</td>
<td align="left">Median :-2.9990</td>
<td align="left">Median :-0.7511</td>
<td align="left">Median :-0.20704</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">Mean : 87.23</td>
<td align="left">NA</td>
<td align="left">Mean :-1.8780</td>
<td align="left">Mean :-0.6306</td>
<td align="left">Mean :-0.13142</td>
</tr>
<tr class="odd">
<td align="left"></td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">3rd Qu.:124.25</td>
<td align="left">NA</td>
<td align="left">3rd Qu.:-0.1243</td>
<td align="left">3rd Qu.: 0.4623</td>
<td align="left">3rd Qu.: 0.05158</td>
</tr>
<tr class="even">
<td align="left"></td>
<td align="left">NA</td>
<td align="left">NA</td>
<td align="left">Max. :169.00</td>
<td align="left">NA</td>
<td align="left">Max. : 3.3221</td>
<td align="left">Max. : 2.3773</td>
<td align="left">Max. : 1.98128</td>
</tr>
</tbody>
</table>
<p>We can also get a summary of the continuous expression variables by status groups. Below, we used ddply function to group the data by status and calculate count,mean and sd of the 3 genes in each group. As shown in the output data.frame, the count column indicates that there are 32 samples in non-recur and 28 in recur group respectively.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="an-overview-of-regression-with-toy-examples.html#cb7-1" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">ddply</span>(toy_data, <span class="sc">~</span>status, summarise, <span class="at">count=</span><span class="fu">length</span>(gene_HOXB13), <span class="at">mean=</span><span class="fu">mean</span>(gene_HOXB13), <span class="at">sd=</span><span class="fu">sd</span>(gene_HOXB13)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">status</th>
<th align="right">count</th>
<th align="right">mean</th>
<th align="right">sd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Status=non-recur</td>
<td align="right">32</td>
<td align="right">-2.8793163</td>
<td align="right">1.243989</td>
</tr>
<tr class="even">
<td align="left">Status=recur</td>
<td align="right">28</td>
<td align="right">-0.7337302</td>
<td align="right">2.320517</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="an-overview-of-regression-with-toy-examples.html#cb8-1" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">ddply</span>(toy_data, <span class="sc">~</span>status, summarise, <span class="at">count=</span><span class="fu">length</span>(gene_IL17BR), <span class="at">mean=</span><span class="fu">mean</span>(gene_IL17BR), <span class="at">sd=</span><span class="fu">sd</span>(gene_IL17BR)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">status</th>
<th align="right">count</th>
<th align="right">mean</th>
<th align="right">sd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Status=non-recur</td>
<td align="right">32</td>
<td align="right">-0.0476809</td>
<td align="right">1.084837</td>
</tr>
<tr class="even">
<td align="left">Status=recur</td>
<td align="right">28</td>
<td align="right">-1.2968660</td>
<td align="right">1.459526</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="an-overview-of-regression-with-toy-examples.html#cb9-1" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">ddply</span>(toy_data, <span class="sc">~</span>status, summarise, <span class="at">count=</span><span class="fu">length</span>(gene_NHLH1), <span class="at">mean=</span><span class="fu">mean</span>(gene_NHLH1), <span class="at">sd=</span><span class="fu">sd</span>(gene_NHLH1)))</span></code></pre></div>
<table>
<thead>
<tr class="header">
<th align="left">status</th>
<th align="right">count</th>
<th align="right">mean</th>
<th align="right">sd</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Status=non-recur</td>
<td align="right">32</td>
<td align="right">-0.1012933</td>
<td align="right">0.6891341</td>
</tr>
<tr class="even">
<td align="left">Status=recur</td>
<td align="right">28</td>
<td align="right">-0.1658441</td>
<td align="right">0.3837488</td>
</tr>
</tbody>
</table>
<p>To prepare for logistic regression, we will create a censor variable as an indicator of the status group, with 1=recur and 0=non-recur. Later, we will use this binary censor variable and the last three columns of continuous expression variables of different genes to perform the analysis.</p>
<p>The codes below show you how to recode status into a binary variable censor. First, we get the index number of the recurrent status, and save them in index object. Then we create a new variable called censor and assign a sequence of 0 to this vector. Next, we extract from censor the elements with index number in index, and assign/recode them as 1. After this recoding finishes, we again take a look at the first 6 rows with head function.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="an-overview-of-regression-with-toy-examples.html#cb10-1" tabindex="-1"></a>index <span class="ot">&lt;-</span> <span class="fu">which</span>(toy_data<span class="sc">$</span>status<span class="sc">==</span><span class="st">&quot;Status=recur&quot;</span>)</span>
<span id="cb10-2"><a href="an-overview-of-regression-with-toy-examples.html#cb10-2" tabindex="-1"></a>toy_data<span class="sc">$</span>censor <span class="ot">&lt;-</span> <span class="fu">rep</span>(<span class="dv">0</span>,<span class="dv">60</span>)</span>
<span id="cb10-3"><a href="an-overview-of-regression-with-toy-examples.html#cb10-3" tabindex="-1"></a>toy_data<span class="sc">$</span>censor[index] <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb10-4"><a href="an-overview-of-regression-with-toy-examples.html#cb10-4" tabindex="-1"></a><span class="fu">kable</span>(<span class="fu">head</span>(toy_data))</span></code></pre></div>
<table>
<colgroup>
<col width="9%" />
<col width="17%" />
<col width="5%" />
<col width="25%" />
<col width="12%" />
<col width="12%" />
<col width="11%" />
<col width="7%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">sampleID</th>
<th align="left">status</th>
<th align="right">time</th>
<th align="left">ratio_group</th>
<th align="right">gene_HOXB13</th>
<th align="right">gene_IL17BR</th>
<th align="right">gene_NHLH1</th>
<th align="right">censor</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">GSM22365</td>
<td align="left">Status=recur</td>
<td align="right">75</td>
<td align="left">High HOXB13:IL17BR ratio</td>
<td align="right">-2.8968509</td>
<td align="right">-2.4744445</td>
<td align="right">-0.1214872</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">GSM22366</td>
<td align="left">Status=recur</td>
<td align="right">21</td>
<td align="left">High HOXB13:IL17BR ratio</td>
<td align="right">-2.9811459</td>
<td align="right">-2.1587305</td>
<td align="right">-0.6215178</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">GSM22367</td>
<td align="left">Status=recur</td>
<td align="right">38</td>
<td align="left">Low HOXB13:IL17BR ratio</td>
<td align="right">-1.3201379</td>
<td align="right">1.0535341</td>
<td align="right">0.0583918</td>
<td align="right">1</td>
</tr>
<tr class="even">
<td align="left">GSM22368</td>
<td align="left">Status=recur</td>
<td align="right">43</td>
<td align="left">High HOXB13:IL17BR ratio</td>
<td align="right">0.5141317</td>
<td align="right">-1.7643865</td>
<td align="right">0.0073468</td>
<td align="right">1</td>
</tr>
<tr class="odd">
<td align="left">GSM22369</td>
<td align="left">Status=non-recur</td>
<td align="right">169</td>
<td align="left">Low HOXB13:IL17BR ratio</td>
<td align="right">-3.0522629</td>
<td align="right">0.8341051</td>
<td align="right">-0.5538062</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">GSM22370</td>
<td align="left">Status=recur</td>
<td align="right">42</td>
<td align="left">Low HOXB13:IL17BR ratio</td>
<td align="right">-3.2735509</td>
<td align="right">-0.9047099</td>
<td align="right">0.8428828</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div id="logistic-regression" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> Logistic Regression<a href="an-overview-of-regression-with-toy-examples.html#logistic-regression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below is a simple case/control study on examining relationship between recurrence status and gene expression using logistic regression.</p>
<p>So what is logistic regression?</p>
<p>Logistic regression is the regression model to conduct when the dependent variable is dichotomous (binary). In this toy example, the binary dependent variable is recur/non-recur status. Similar to other regression methods, the logistic regression is a predictive analysis but it also reveals the association relationship between the dependent variable and the independent variables.</p>
<p>Logistic regression can help us answer a variety of questions, including, but not limited to the examples below:</p>
<ul>
<li>How does the probability of getting lung cancer (yes vs. no) change for every additional pound of overweight and for every pack of cigarettes smoked per day?</li>
<li>Do body weight, calorie intake, fat intake, and participant age have an influence on heart attacks (yes vs. no)?
Mathematically, the logistic regression equation can be defined as:</li>
</ul>
<p><span class="math inline">\(l = log(\frac{p}{1-p}) = \beta_0 + \beta_1 X_1 + \beta_2 X_2 + ...\)</span>, where <span class="math inline">\(p\)</span> is the probability that the response <span class="math inline">\(Y = 1\)</span>.</p>
<p>More detailed explanation about logistic regression can be found at: <a href="https://en.wikipedia.org/wiki/Logistic_regression" class="uri">https://en.wikipedia.org/wiki/Logistic_regression</a></p>
<p>In the code below, we performed a logistic regression with a binary response <code>censor</code> and continuous expression of <code>gene_HOXB13</code>. The model is fitted with function <code>glm</code>. <code>cencor~gene_HOXB13</code> specifies the model structure, where <code>censor</code> is the response and <code>gene_HOCB13</code> is the explanatory variable. <code>family</code> argument specifies the distribution of <code>censor</code> as “binomial”, and link specifies that link function for binomial response is “logit”, which is why this model is called “logistic model”.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="an-overview-of-regression-with-toy-examples.html#cb11-1" tabindex="-1"></a><span class="co"># logistic regresion, use gene HOXB13 to predict the recur/non-recur status</span></span>
<span id="cb11-2"><a href="an-overview-of-regression-with-toy-examples.html#cb11-2" tabindex="-1"></a><span class="co"># for significant gene: HOXB13</span></span>
<span id="cb11-3"><a href="an-overview-of-regression-with-toy-examples.html#cb11-3" tabindex="-1"></a>fit.logistic<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">glm</span>(censor<span class="sc">~</span> gene_HOXB13,<span class="at">data =</span> toy_data,<span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">&#39;logit&#39;</span>))</span>
<span id="cb11-4"><a href="an-overview-of-regression-with-toy-examples.html#cb11-4" tabindex="-1"></a><span class="fu">summary</span>(fit.logistic<span class="fl">.1</span>) <span class="co"># output summary of the logistic regression</span></span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = censor ~ gene_HOXB13, family = binomial(link = &quot;logit&quot;), 
##     data = toy_data)
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)   1.1904     0.5474   2.175 0.029655 *  
## gene_HOXB13   0.6440     0.1913   3.366 0.000764 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 82.911  on 59  degrees of freedom
## Residual deviance: 65.216  on 58  degrees of freedom
## AIC: 69.216
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>The table under Coefficients contains test results. The p-value of the gene HOXB13, in the last column of this table, is 0.000764 which is way smaller than 0.05. It indicates that this gene has a significant impact on the recurrence status.</p>
</div>
<div id="roc-curve-and-auc-for-hoxb13" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> ROC curve and AUC for HOXB13<a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The ROC (receiver operating characteristic) curve and AUC (area under the curve) are popular tools to check the predictive ability of the logistic regression.</p>
<p>Roughly speaking, the AUC equals to the probability that a classifier will rank a randomly chosen positive instance higher than a randomly chosen negative one. AUC is always between 0 and 1. The closer to 1 the AUC is, the better the model fits. As shown in the following results, the AUC is 0.796 and pretty close to 1. So this model fits the data well. In other words, we can say the expression of gene HOXB13 has a great explanation power on the recur status.</p>
<p>For more information about these two concepts, Please visit: <a href="https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve" class="uri">https://en.wikipedia.org/wiki/Receiver_operating_characteristic#Area_under_the_curve</a></p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="an-overview-of-regression-with-toy-examples.html#cb13-1" tabindex="-1"></a><span class="co">#plot ROC curve</span></span>
<span id="cb13-2"><a href="an-overview-of-regression-with-toy-examples.html#cb13-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit.logistic<span class="fl">.1</span>, <span class="at">type=</span><span class="st">&quot;response&quot;</span>)</span>
<span id="cb13-3"><a href="an-overview-of-regression-with-toy-examples.html#cb13-3" tabindex="-1"></a>pr <span class="ot">&lt;-</span> <span class="fu">prediction</span>(p, toy_data<span class="sc">$</span>censor)</span>
<span id="cb13-4"><a href="an-overview-of-regression-with-toy-examples.html#cb13-4" tabindex="-1"></a>prf <span class="ot">&lt;-</span> <span class="fu">performance</span>(pr, <span class="at">measure =</span> <span class="st">&quot;tpr&quot;</span>, <span class="at">x.measure =</span> <span class="st">&quot;fpr&quot;</span>)</span>
<span id="cb13-5"><a href="an-overview-of-regression-with-toy-examples.html#cb13-5" tabindex="-1"></a>prf1 <span class="ot">&lt;-</span> prf</span>
<span id="cb13-6"><a href="an-overview-of-regression-with-toy-examples.html#cb13-6" tabindex="-1"></a><span class="fu">plot</span>(prf1,<span class="at">main=</span><span class="st">&quot;ROC for significant gene HOXB13&quot;</span>)</span></code></pre></div>
<p><img src="cshl_2024_files/figure-html/unnamed-chunk-10-1.png" width="672" /></p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="an-overview-of-regression-with-toy-examples.html#cb14-1" tabindex="-1"></a><span class="co"># calculate the auc</span></span>
<span id="cb14-2"><a href="an-overview-of-regression-with-toy-examples.html#cb14-2" tabindex="-1"></a>auc <span class="ot">&lt;-</span> <span class="fu">performance</span>(pr, <span class="at">measure =</span> <span class="st">&quot;auc&quot;</span>)</span>
<span id="cb14-3"><a href="an-overview-of-regression-with-toy-examples.html#cb14-3" tabindex="-1"></a>auc <span class="ot">&lt;-</span> auc<span class="sc">@</span>y.values[[<span class="dv">1</span>]]</span>
<span id="cb14-4"><a href="an-overview-of-regression-with-toy-examples.html#cb14-4" tabindex="-1"></a>auc <span class="co"># 0.796</span></span></code></pre></div>
<pre><code>## [1] 0.7957589</code></pre>
</div>
<div id="practice" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Practice<a href="an-overview-of-regression-with-toy-examples.html#practice" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Practice 1: Logistic regression</strong>
We can repeat the same analysis using another gene, namely NHLH1. What can you conclude from the test results?</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="an-overview-of-regression-with-toy-examples.html#cb16-1" tabindex="-1"></a><span class="co"># for non-significant gene NHLH1</span></span>
<span id="cb16-2"><a href="an-overview-of-regression-with-toy-examples.html#cb16-2" tabindex="-1"></a>fit.logistic<span class="fl">.2</span> <span class="ot">&lt;-</span> <span class="fu">glm</span>(censor<span class="sc">~</span> gene_NHLH1,<span class="at">data =</span> toy_data,<span class="at">family =</span> <span class="fu">binomial</span>(<span class="at">link =</span> <span class="st">&#39;logit&#39;</span>))</span>
<span id="cb16-3"><a href="an-overview-of-regression-with-toy-examples.html#cb16-3" tabindex="-1"></a><span class="fu">summary</span>(fit.logistic<span class="fl">.2</span>)</span></code></pre></div>
<pre><code>## 
## Call:
## glm(formula = censor ~ gene_NHLH1, family = binomial(link = &quot;logit&quot;), 
##     data = toy_data)
## 
## Coefficients:
##             Estimate Std. Error z value Pr(&gt;|z|)
## (Intercept)  -0.1616     0.2672  -0.605    0.545
## gene_NHLH1   -0.2089     0.4698  -0.445    0.657
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 82.911  on 59  degrees of freedom
## Residual deviance: 82.710  on 58  degrees of freedom
## AIC: 86.71
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p><strong>Practice 2: ROC curve and AUC for NHLH1</strong>
We draw the same ROC curve and calculate AUC for NHLH1, what can you conclude from results for NHLH1?</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="an-overview-of-regression-with-toy-examples.html#cb18-1" tabindex="-1"></a><span class="co">#plot ROC curve</span></span>
<span id="cb18-2"><a href="an-overview-of-regression-with-toy-examples.html#cb18-2" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">predict</span>(fit.logistic<span class="fl">.2</span>, <span class="at">type=</span><span class="st">&quot;response&quot;</span>)</span>
<span id="cb18-3"><a href="an-overview-of-regression-with-toy-examples.html#cb18-3" tabindex="-1"></a>pr <span class="ot">&lt;-</span> <span class="fu">prediction</span>(p, toy_data<span class="sc">$</span>censor)</span>
<span id="cb18-4"><a href="an-overview-of-regression-with-toy-examples.html#cb18-4" tabindex="-1"></a>prf <span class="ot">&lt;-</span> <span class="fu">performance</span>(pr, <span class="at">measure =</span> <span class="st">&quot;tpr&quot;</span>, <span class="at">x.measure =</span> <span class="st">&quot;fpr&quot;</span>)</span>
<span id="cb18-5"><a href="an-overview-of-regression-with-toy-examples.html#cb18-5" tabindex="-1"></a>prf2 <span class="ot">&lt;-</span> prf</span>
<span id="cb18-6"><a href="an-overview-of-regression-with-toy-examples.html#cb18-6" tabindex="-1"></a><span class="fu">plot</span>(prf2,<span class="at">main=</span><span class="st">&quot;ROC for non-significant gene NHLH1 &quot;</span>)</span></code></pre></div>
<p><img src="cshl_2024_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="an-overview-of-regression-with-toy-examples.html#cb19-1" tabindex="-1"></a><span class="co"># calculate the auc</span></span>
<span id="cb19-2"><a href="an-overview-of-regression-with-toy-examples.html#cb19-2" tabindex="-1"></a>auc <span class="ot">&lt;-</span> <span class="fu">performance</span>(pr, <span class="at">measure =</span> <span class="st">&quot;auc&quot;</span>)</span>
<span id="cb19-3"><a href="an-overview-of-regression-with-toy-examples.html#cb19-3" tabindex="-1"></a>auc <span class="ot">&lt;-</span> auc<span class="sc">@</span>y.values[[<span class="dv">1</span>]]</span>
<span id="cb19-4"><a href="an-overview-of-regression-with-toy-examples.html#cb19-4" tabindex="-1"></a>auc <span class="co"># 0.47</span></span></code></pre></div>
<pre><code>## [1] 0.4709821</code></pre>
</div>
<div id="comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve" class="section level2 hasAnchor" number="2.6">
<h2><span class="header-section-number">2.6</span> Comparison of HOXB13 ROC curve and NHLH1 ROC curve<a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To have a direct compare the ROC curve for these 2 genes, we can plot the two ROC curves on a single plot. The code below generate the combined plot. We first plot ROC curve for HOXB13 as normal with color blue, then add the ROC curve of NHLH1 on this plot using lines and color red. legend function specifies the plot legend which is shown at the bottom of the plot.</p>
<p>With this combined plot, we can conclude that HOXB13 has much higher predictive probability than NHLH1, while retaining a low false positive rate.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="an-overview-of-regression-with-toy-examples.html#cb21-1" tabindex="-1"></a><span class="do">### two ROC curve comparison</span></span>
<span id="cb21-2"><a href="an-overview-of-regression-with-toy-examples.html#cb21-2" tabindex="-1"></a><span class="fu">plot</span>(prf1,<span class="at">main=</span><span class="st">&quot;ROC for HOXB13 and NHLH1&quot;</span>,<span class="at">col=</span><span class="st">&#39;blue&#39;</span>)</span>
<span id="cb21-3"><a href="an-overview-of-regression-with-toy-examples.html#cb21-3" tabindex="-1"></a><span class="fu">lines</span>(prf2<span class="sc">@</span>x.values[[<span class="dv">1</span>]],prf2<span class="sc">@</span>y.values[[<span class="dv">1</span>]],<span class="at">col=</span><span class="st">&#39;red&#39;</span>)</span>
<span id="cb21-4"><a href="an-overview-of-regression-with-toy-examples.html#cb21-4" tabindex="-1"></a><span class="fu">legend</span>(<span class="st">&#39;bottomright&#39;</span>,<span class="at">lwd=</span><span class="dv">2</span>,</span>
<span id="cb21-5"><a href="an-overview-of-regression-with-toy-examples.html#cb21-5" tabindex="-1"></a>       <span class="at">legend=</span><span class="fu">c</span>(<span class="st">&quot;ROC for significant gene HOXB13 (auc=0.796)&quot;</span>,<span class="st">&quot;ROC for non-significant gene NHLH1 (auc =0.47)&quot;</span>),</span>
<span id="cb21-6"><a href="an-overview-of-regression-with-toy-examples.html#cb21-6" tabindex="-1"></a>       <span class="at">col=</span><span class="fu">c</span>(<span class="st">&#39;blue&#39;</span>,<span class="st">&#39;red&#39;</span>))</span></code></pre></div>
<p><img src="cshl_2024_files/figure-html/unnamed-chunk-13-1.png" width="672" /></p>
</div>
<div id="linear-regression" class="section level2 hasAnchor" number="2.7">
<h2><span class="header-section-number">2.7</span> Linear Regression<a href="an-overview-of-regression-with-toy-examples.html#linear-regression" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In statistics, linear regression is an approach for modeling the relationship between a continuous dependent variable Y and one or more explanatory variables (or independent variables) denoted by X. The explanatory variables can be either continuous (e.g. age, weight) or categorical (e.g. gender, smoke or not).</p>
<p>In the following example, we examine the relationship between the expression of gene HOXB13 and IL17BR. Both of them are continuous variables.</p>
<p>For more detailed explanation of linear regression, please visit: <a href="https://en.wikipedia.org/wiki/Linear_regression" class="uri">https://en.wikipedia.org/wiki/Linear_regression</a></p>
<div id="linear-regression-of-hoxb13-vs.-il17br" class="section level3 hasAnchor" number="2.7.1">
<h3><span class="header-section-number">2.7.1</span> Linear regression of HOXB13 vs. IL17BR<a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Linear regression is fitted with lm function. The arguments in lm is very similar to those in glm, but much simpler as we don’t need to specify the distribution and link function here.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="an-overview-of-regression-with-toy-examples.html#cb22-1" tabindex="-1"></a><span class="co"># linear model, use gene IL17BR to predict another gene HOXB13</span></span>
<span id="cb22-2"><a href="an-overview-of-regression-with-toy-examples.html#cb22-2" tabindex="-1"></a>fit.lm <span class="ot">&lt;-</span> <span class="fu">lm</span>(gene_HOXB13<span class="sc">~</span>gene_IL17BR,<span class="at">data =</span> toy_data)</span>
<span id="cb22-3"><a href="an-overview-of-regression-with-toy-examples.html#cb22-3" tabindex="-1"></a><span class="fu">summary</span>(fit.lm) <span class="co"># print the summary of the fitted regression</span></span></code></pre></div>
<pre><code>## 
## Call:
## lm(formula = gene_HOXB13 ~ gene_IL17BR, data = toy_data)
## 
## Residuals:
##     Min      1Q  Median      3Q     Max 
## -2.5037 -1.5191 -0.6175  1.1574  5.3354 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)  -2.1772     0.2855  -7.626  2.6e-10 ***
## gene_IL17BR  -0.4743     0.1861  -2.548   0.0135 *  
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 2.016 on 58 degrees of freedom
## Multiple R-squared:  0.1007, Adjusted R-squared:  0.08519 
## F-statistic: 6.494 on 1 and 58 DF,  p-value: 0.01349</code></pre>
<p>The p-value for IL17BR is 0.0135&lt;0.05, so there is a significant association between expression of HOXB13 and IL17BR.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/01-intro.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_2024.pdf", "cshl_2024.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
