<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2 | 2024 CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2 | 2024 CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  <meta name="github-repo" content="rstudio/cshl_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2 | 2024 CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  

<meta name="author" content="Danni Liu - Purdue University" />


<meta name="date" content="2024-06-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="an-overview-of-regression-with-toy-examples.html"/>
<link rel="next" href="an-imputation-example.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.4</b> An Example of Summary Statistics</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#input-files"><i class="fa fa-check"></i><b>3.4.1</b> Input files</a></li>
<li class="chapter" data-level="3.4.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-missing-rate"><i class="fa fa-check"></i><b>3.4.2</b> Calculating missing rate</a></li>
<li class="chapter" data-level="3.4.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-allel-frequencies"><i class="fa fa-check"></i><b>3.4.3</b> Calculating allel frequencies</a></li>
<li class="chapter" data-level="3.4.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>3.4.4</b> Hardy-Weinberg Equilibrium</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.5</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.6</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="an-imputation-example.html"><a href="an-imputation-example.html"><i class="fa fa-check"></i><b>4</b> An Imputation Example</a></li>
<li class="chapter" data-level="5" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html"><i class="fa fa-check"></i><b>5</b> Population Stratification Correction Using EIGENSTRAT</a>
<ul>
<li class="chapter" data-level="5.1" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html#eigenstrat-input-files"><i class="fa fa-check"></i><b>5.1</b> EIGENSTRAT input files</a></li>
<li class="chapter" data-level="5.2" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html#run-eigenstrat-using-input-genotype-data"><i class="fa fa-check"></i><b>5.2</b> Run EIGENSTRAT Using Input Genotype Data</a></li>
<li class="chapter" data-level="5.3" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>5.3</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="simple-association-test.html"><a href="simple-association-test.html"><i class="fa fa-check"></i><b>6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="6.1" data-path="simple-association-test.html"><a href="simple-association-test.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="6.2" data-path="simple-association-test.html"><a href="simple-association-test.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
<li class="chapter" data-level="6.3" data-path="simple-association-test.html"><a href="simple-association-test.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>6.3</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="simple-association-test.html"><a href="simple-association-test.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>6.3.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="6.3.2" data-path="simple-association-test.html"><a href="simple-association-test.html#residual-phenotype-data-after-covariates-adjustments-in-unix"><i class="fa fa-check"></i><b>6.3.2</b> Residual phenotype data after covariates adjustments (in UNIX):</a></li>
<li class="chapter" data-level="6.3.3" data-path="simple-association-test.html"><a href="simple-association-test.html#association-test-using-plink"><i class="fa fa-check"></i><b>6.3.3</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html"><i class="fa fa-check"></i><b>7</b> Unit 3. GWAS Case Study-Continuous Trait</a>
<ul>
<li class="chapter" data-level="7.1" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html#environment-setting-and-packages-1"><i class="fa fa-check"></i><b>7.1</b> Environment setting and packages</a></li>
<li class="chapter" data-level="7.2" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html#accessing-data"><i class="fa fa-check"></i><b>7.2</b> Accessing data</a>
<ul>
<li class="chapter" data-level="7.2.1" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html#data-availability"><i class="fa fa-check"></i><b>7.2.1</b> Data availability</a></li>
<li class="chapter" data-level="7.2.2" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html#data-files-documentation"><i class="fa fa-check"></i><b>7.2.2</b> Data files documentation</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>8</b> Applications</a>
<ul>
<li class="chapter" data-level="8.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>8.1</b> Example one</a></li>
<li class="chapter" data-level="8.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>8.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>9</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">2024 CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This notebook demonstrates a standard approach to conduct the genome wide association study (GWAS) using toy example datasets. The procedures consist of four steps:</p>
<ul>
<li>Quality Control</li>
<li>Imputation</li>
<li>Population Stratification Correction</li>
<li>Simple Association Test
The data used for this example is a toy genotype data containing 5 random SNPs and a random phenotype data with covariates.</li>
</ul>
<p>We use both bash and R codes in this notebook. The type of programming language is labeled at the top of each code cell.</p>
<blockquote>
<p><span style="color: #0064a4;"><br />
Special notes for Windows users: Windows has two command-line shells, Command shell and Windows Powershell. Command shell is specific to Windows system and can be run from RStudio Terminal or in “Command Prompt” App, but the language it uses is NOT the same as Bash! Powershell App is the only place to run Bash commands in Windows. Therefore if you are Windows users and use RStudio for this course, you can’t run bash codes such as “head” or “awk” from RStudio Terminal, you can <strong>ONLY</strong> run PLINK program codes. If you would like to try Bash codes, please open “Windows Powershell” app and follow the same steps to set the paths. Still, Powershell App won’t run PLINK codes, so you’ll have to switch between Powershell and RStudio Terminal.
</span></p>
</blockquote>
<div id="environment-setting" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Environment Setting<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#environment-setting" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before we can run any analysis, we need to set the working directory path to the downloaded “bigcare-main” folder for both Terminal and Console in RStudio. Below are the codes to setup paths at both places. These paths have to be setup correctly at first, otherwise we can’t locate the correct input files.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-1" tabindex="-1"></a><span class="co"># change the paths to where you saved &quot;bigcare-main&quot;</span></span>
<span id="cb1-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-2" tabindex="-1"></a><span class="do">## run in Terminal</span></span>
<span id="cb1-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-3" tabindex="-1"></a>mydir<span class="ot">=</span><span class="st">&quot;/path/to/cshl_2024&quot;</span></span>
<span id="cb1-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-4" tabindex="-1"></a>cd <span class="sc">$</span>mydir</span>
<span id="cb1-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-5" tabindex="-1"></a></span>
<span id="cb1-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-6" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb1-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb1-7" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/path/to/cshl_2024&quot;</span>)</span></code></pre></div>
</div>
<div id="gwas-tools" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> GWAS Tools<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#gwas-tools" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will use two main program tools in this notebook, PLINK and IMPUTE2.</p>
<p>PLINK is used for quality control and perform association test for large scale genotype data. You can download PLINK program using this link. Download the .zip file or find the .zip files suitable for your laptop system in “bigcare-main/Tools/PLINK” and unzip it to call the plink program in the folder.</p>
<p>IMPUTE2 is used for genotype imputation and haplotype phasing. You can download its program using this link. IMPUTE2 program is only available to run in Linux or macOS system. You can find the zipped installation file in “bigcare-main/Tools/IMPUTE2”. If you want to perform imputation for your own data, you also need to download the required reference genome from the above link. But in this notebook will only use the example genome.</p>
<blockquote>
<p><span style="color: #0064a4;">
Important: If you want to run these two programs on your laptop, please make sure you have unzipped the installation packages in “./bigcare-main/Tools” that is suitable for your laptop systerm. We will directly call PLINK and IMPUTE2 programs using relative path.
</span></p>
</blockquote>
<p>Instructions on installing these two tools can be found on course GitHub here.</p>
<p>For example, my computer is macOS system so I can run both PLINK and IMPUTE2. After I downloaded “bigcare-main” folder, I open ./Tools/PLINK and unzip the plink package for macOS: “plink_mac_20230116.zip”. A folder “plink_mac_20230116” will be created which contains a program file called “plink”. Then, I can call PLINK program by specifying the absolute path of this “plink” program: /path/to/bigcare-main/Tools/PLINK/plink_mac_20230116/plink. Similar for Windows system, you need to unzip the package “plink_win64_20230116.zip” with right click-&gt;“Extract All…”. Then you can call PLINK with /path/to/bigcare-main/Tools/PLINK/plink_win64_20230116/plink.exe.</p>
<p>Similar for IMPUTE2, I unzip “impute_v2.3.2_MacOSX_Intel.tgz” for macOS under “./bigcare-main/Tools/IMPUTE2/”. And I can call IMPUTE2 program with /path/to/bigcare-main/Tools/IMPUTE2/impute_v2.3.2_MacOSX_Intel/impute2. Note that Windows system won’t be able to run IMPUTE2.</p>
<p>You should modify the above /path/to/bigcare to where your “bigcare-main” folder is.</p>
<blockquote>
<p><span style="color: #0064a4;">
However, macOS users need to go through additional steps to authorize PLINK and IMPUTE2 programs before you can call them! See below.
</span></p>
</blockquote>
<p>When you try to call PLINK or IMPUTE2 with the above commands, you might get a pop-up window with the following message “plink” cannot be opened because the developer cannot be verified.. Select “Cancel” and we will authorize PLINK first. Please go to macOS “System Settings”-&gt;“Privacy &amp; Security”, and scroll down until you see a message “plink” was blocked from use because it is not from an identified developer.. Click “Allow Anyway” and you will be prompt to enter your Mac username and password. Click “Unlock” and then “Allow Anyway” again, and the previous message will disappear. Then try to run the same codes again in RStudio Terminal. You will get another pop-up window saying macOS cannot verify the developer of “plink”. Are you sure you want to open it?, select “Open” then you can see PLINK or IMPUTE2 commands options show up in Terminal.</p>
</div>
<div id="quality-control" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Quality Control<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#quality-control" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>PLINK will generate a number of standard summary statistics that are useful for quality control (e.g. missing genotype rate, minor allele frequency, Hardy-Weinberg equilibrium test and non-Mendelian transmission rate). These can also be used as thresholds for subsequent analyses. For more detailed information about plink, please refer its website <a href="http://zzz.bwh.harvard.edu/plink/" class="uri">http://zzz.bwh.harvard.edu/plink/</a>.</p>
<p>The following is a list of some common options that can be used to filter out individuals or SNPs on the basis of the summary statistic measures:</p>
<table>
<colgroup>
<col width="37%" />
<col width="31%" />
<col width="31%" />
</colgroup>
<thead>
<tr class="header">
<th>Feature inclusion criteria</th>
<th>As summary statistic</th>
<th>As inclusion criteria</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Missingness per individual</td>
<td><code>--missing</code></td>
<td><code>--mind N</code>.</td>
</tr>
<tr class="even">
<td>Missingness per marker</td>
<td>–missing</td>
<td>–geno N</td>
</tr>
<tr class="odd">
<td>Allele frequency</td>
<td>–freq</td>
<td>–maf N</td>
</tr>
<tr class="even">
<td>Hardy-Weinberg equilibrium</td>
<td>–hardy</td>
<td>–hwe N</td>
</tr>
</tbody>
</table>
</div>
<div id="an-example-of-summary-statistics" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> An Example of Summary Statistics<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-summary-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="input-files" class="section level3 hasAnchor" number="3.4.1">
<h3><span class="header-section-number">3.4.1</span> Input files<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#input-files" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Input files: in folder <code>./QC</code>– We will use a random selected 5 SNPs as an example data:</p>
<ul>
<li>5SNPs.ped</li>
<li>5SNPs.map</li>
</ul>
<p>The PED file is a white-space delimited file: the first six columns are mandatory:</p>
<pre><code> Family ID
 Individual ID
 Paternal ID
 Maternal ID
 Sex (1=male; 2=female; other=unknown)
 Phenotype</code></pre>
<div class="sourceCode" id="cb3"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb3-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb3-1" tabindex="-1"></a><span class="co"># bash codes- run in Terminal</span></span>
<span id="cb3-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb3-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App </span></span>
<span id="cb3-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb3-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/5SNPs.ped</span></code></pre></div>
<pre><code>## 1 206-001 0 0 1 0 A A G G G G G G A A
## 2 206-003 0 0 1 0 A A G G G G G G A A
## 3 206-004 0 0 1 0 A A G G G G G G A A
## 4 206-014 0 0 1 0 A A G G G G G G A A
## 5 206-015 0 0 1 0 A G A G A G A G A A
## 6 206-016 0 0 1 0 A G A G A G A G A A
## 7 206-018 0 0 1 0 A A G G G G G G A A
## 8 206-019 0 0 1 0 A A A G A G A G A G
## 9 206-021 0 0 1 0 A A G G G G G G A A
## 10 206-023 0 0 1 0 A G A G A G A G A A</code></pre>
<p>By default, each line of the MAP file describes a single marker and must contain exactly 4 columns:</p>
<pre><code> chromosome (1-22, X, Y or 0 if unplaced)
 rs# or snp identifier
 Genetic distance (morgans)
 Base-pair position (bp units)</code></pre>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb6-1" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/5SNPs.map</span></code></pre></div>
<pre><code>## 1 rs3094315 0 752566
## 1 rs2073813 0 753541
## 1 rs3131969 0 754182
## 1 rs3131967 0 754334
## 1 rs77598327 0 757120</code></pre>
<p>Three types of summary statistics that we are going to calculate using <code>PLINK</code>:</p>
<ol style="list-style-type: decimal">
<li>missingness checking</li>
<li>allele frequencies</li>
<li>hardy-weinberg</li>
</ol>
<p>All results generated from the following commands are saved in folder <code>./Data/QC</code>. You can open the files in this folder to view the data structure.</p>
</div>
<div id="calculating-missing-rate" class="section level3 hasAnchor" number="3.4.2">
<h3><span class="header-section-number">3.4.2</span> Calculating missing rate<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-missing-rate" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following commands compute the <strong>missingness</strong> for each SNP and each subject. (The output will print all calculation progress generated by <code>PLINK</code>.)</p>
<ol style="list-style-type: decimal">
<li><strong>–noweb</strong>, suppress the connection to internet for version check.</li>
<li><strong>–file</strong>, specify the input files.</li>
<li><strong>–missing</strong>, calculate the number of missing and missing rate.</li>
<li><strong>–out</strong>, specify the output file name.</li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb8-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb8-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb8-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb8-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb8-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb8-3" tabindex="-1"></a></span>
<span id="cb8-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb8-4" tabindex="-1"></a><span class="co"># for windows users</span></span>
<span id="cb8-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb8-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64/plink.exe</span> <span class="dt">\</span></span>
<span id="cb8-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb8-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--missing</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb9-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb9-1" tabindex="-1"></a><span class="co"># for macoS users</span></span>
<span id="cb9-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb9-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac/plink</span> <span class="dt">\</span></span>
<span id="cb9-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb9-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--missing</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90p 64-bit (11 Dec 2023)              www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --missing
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## --missing: Sample missing data report written to ./Data/QC/younameit.imiss, and
## variant-based missing data report written to ./Data/QC/younameit.lmiss.</code></pre>
<p>This option creates two output files in ‘results’ folder:</p>
<ol style="list-style-type: decimal">
<li><strong>“younameit.imiss”</strong>, missingness by individual.</li>
<li><strong>“younameit.lmiss”</strong>, missingness by SNP.</li>
</ol>
<p>For individuals “.imiss” file, the format is:</p>
<pre><code> FID                Family ID
 IID                Individual ID
 MISS_PHENO         Missing phenotype? (Y/N)
 N_MISS             Number of missing SNPs
 N_GENO             Number of non-obligatory missing genotypes
 F_MISS             Proportion of missing SNPs</code></pre>
<div class="sourceCode" id="cb12"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb12-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb12-1" tabindex="-1"></a><span class="co"># bash codes- run in Terminal</span></span>
<span id="cb12-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb12-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App </span></span>
<span id="cb12-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb12-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.imiss</span></code></pre></div>
<pre><code>##  FID       IID MISS_PHENO   N_MISS   N_GENO   F_MISS
##    1   206-001          Y        0        5        0
##    2   206-003          Y        0        5        0
##    3   206-004          Y        0        5        0
##    4   206-014          Y        0        5        0
##    5   206-015          Y        0        5        0
##    6   206-016          Y        0        5        0
##    7   206-018          Y        0        5        0
##    8   206-019          Y        0        5        0
##    9   206-021          Y        0        5        0</code></pre>
<p>For SNP “lmiss” file, the format is:</p>
<pre><code> SNP                SNP identifier
 CHR                Chromosome number
 N_MISS             Number of individuals missing this SNP
 N_GENO             Number of non-obligatory missing genotypes
 F_MISS             Proportion of sample missing for this SNP</code></pre>
<div class="sourceCode" id="cb15"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb15-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb15-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb15-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb15-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App </span></span>
<span id="cb15-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb15-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.lmiss</span></code></pre></div>
<pre><code>##  CHR         SNP   N_MISS   N_GENO   F_MISS
##    1   rs3094315        2      344 0.005814
##    1   rs2073813       14      344   0.0407
##    1   rs3131969       11      344  0.03198
##    1   rs3131967        3      344 0.008721
##    1  rs77598327       10      344  0.02907</code></pre>
</div>
<div id="calculating-allel-frequencies" class="section level3 hasAnchor" number="3.4.3">
<h3><span class="header-section-number">3.4.3</span> Calculating allel frequencies<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-allel-frequencies" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following command computes the <strong>allele frequencies</strong>. (The output will print all calculation progress generated by <code>PLINK</code>.)
1. <strong>–freq</strong>, calculate the allele frequencies.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb17-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb17-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb17-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb17-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb17-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb17-3" tabindex="-1"></a></span>
<span id="cb17-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb17-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb17-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb17-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64/plink.exe</span> <span class="dt">\</span></span>
<span id="cb17-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb17-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--freq</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb18-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb18-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb18-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb18-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac/plink</span> <span class="dt">\</span></span>
<span id="cb18-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb18-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--freq</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90p 64-bit (11 Dec 2023)              www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --freq
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## --freq: Allele frequencies (founders only) written to ./Data/QC/younameit.frq .</code></pre>
<p>This command will create a output file <strong>“younameit.frq”</strong> with six columns:</p>
<pre><code> CHR       Chromosome
 
 SNP       SNP identifier
 
 A1        Allele 1 code (minor allele)
 
 A2        Allele 2 code (major allele)
 
 MAF       Minor allele frequency
 
 NCHROBS   Non-missing allele count</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb21-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb21-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb21-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb21-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb21-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb21-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.frq</span></code></pre></div>
<pre><code>##  CHR         SNP   A1   A2          MAF  NCHROBS
##    1   rs3094315    G    A      0.07749      684
##    1   rs2073813    A    G       0.2242      660
##    1   rs3131969    A    G       0.2387      666
##    1   rs3131967    A    G       0.2082      682
##    1  rs77598327    G    A       0.1228      668</code></pre>
</div>
<div id="hardy-weinberg-equilibrium" class="section level3 hasAnchor" number="3.4.4">
<h3><span class="header-section-number">3.4.4</span> Hardy-Weinberg Equilibrium<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#hardy-weinberg-equilibrium" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following command computes the <strong>Hardy-Weiberg test</strong>. (The output will print all calculation progress generated by <code>PLINK</code>.)
1. <strong>–hardy</strong>, perform hardy-weinberg test.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb23-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb23-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb23-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb23-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb23-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb23-3" tabindex="-1"></a></span>
<span id="cb23-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb23-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb23-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb23-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64/plink.exe</span> <span class="dt">\</span></span>
<span id="cb23-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb23-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--hardy</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb24-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb24-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb24-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb24-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac/plink</span> <span class="dt">\</span></span>
<span id="cb24-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb24-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--hardy</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90p 64-bit (11 Dec 2023)              www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --hardy
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## --hardy: Writing Hardy-Weinberg report (founders only) to
## ./Data/QC/younameit.hwe ... 0%0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.</code></pre>
<p>This command will create an output file <strong>“younameit.hwe”</strong> with the following columns:</p>
<pre><code> SNP             SNP identifier
 
 TEST            Code indicating whether the sample is a case or control
 
 A1              Minor allele code
 
 A2              Major allele code
 
 GENO            Genotype counts: 11/12/22 
 
 O(HET)          Observed heterozygosity
 
 E(HET)          Expected heterozygosity
 
 P               P-value for Hardy-Weinberg test</code></pre>
<div class="sourceCode" id="cb27"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb27-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb27-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb27-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb27-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb27-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb27-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.hwe</span></code></pre></div>
<pre><code>##  CHR         SNP     TEST   A1   A2                 GENO   O(HET)   E(HET)            P 
##    1   rs3094315  ALL(NP)    G    A             3/47/292   0.1374    0.143       0.4398
##    1   rs2073813  ALL(NP)    A    G           18/112/200   0.3394   0.3479       0.6366
##    1   rs3131969  ALL(NP)    A    G           24/111/198   0.3333   0.3635       0.1323
##    1   rs3131967  ALL(NP)    A    G           12/118/211    0.346   0.3297       0.4149
##    1  rs77598327  ALL(NP)    G    A             2/78/254   0.2335   0.2154       0.1984</code></pre>
</div>
</div>
<div id="an-example-of-inclusion-thresholds-for-qc" class="section level2 hasAnchor" number="3.5">
<h2><span class="header-section-number">3.5</span> An Example of Inclusion Thresholds for QC<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-inclusion-thresholds-for-qc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Input files: in folder <code>./QC</code></p>
<ul>
<li><p>5SNPs.ped</p></li>
<li><p>5SNPs.map</p></li>
</ul>
<p>The following command performs <strong>multiple quality controls</strong>. (The output will print all calculation progress generated by <code>PLINK</code>.)
1. <strong>–mind</strong> 0.1, exclude the individuals with missing rate larger than 0.1. You can change 0.1 to other desired values.
2. <strong>–geno</strong> 0.1, exclude the SNPs with missing rate larger than 0.1. You can change 0.1 to other desiered values.
3. <strong>–maf</strong> 0.05, exclude the SNPs with minor allele frequency (MAF) smaller than 0.05. You can change 0.05 to other desiered values.
4. <strong>–hwe</strong> 0.001, exclude the SNPs with P-values of Hardy-Weinberg test smaller than 0.001. You can change it to otehr desired values.
5. <strong>–recode</strong>, generate a new dataset.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb29-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb29-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb29-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-3" tabindex="-1"></a></span>
<span id="cb29-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb29-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64/plink.exe</span> <span class="dt">\</span></span>
<span id="cb29-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="dt">\</span></span>
<span id="cb29-7"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb29-7" tabindex="-1"></a>--mind 0.1 <span class="at">--geno</span> 0.1 <span class="at">--maf</span> 0.05 <span class="at">--hwe</span> 0.001 <span class="at">--recode</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb30"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb30-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb30-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb30-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb30-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac/plink</span> <span class="dt">\</span></span>
<span id="cb30-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb30-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="dt">\</span></span>
<span id="cb30-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb30-4" tabindex="-1"></a>--mind 0.1 <span class="at">--geno</span> 0.1 <span class="at">--maf</span> 0.05 <span class="at">--hwe</span> 0.001 <span class="at">--recode</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90p 64-bit (11 Dec 2023)              www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --geno 0.1
##   --hwe 0.001
##   --maf 0.05
##   --mind 0.1
##   --noweb
##   --out ./Data/QC/younameit
##   --recode
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## 32 people removed due to missing genotype data (--mind).
## IDs written to ./Data/QC/younameit.irem .
## Using 1 thread (no multithreaded calculations invoked).
## Before main variant filters, 312 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate in remaining samples is exactly 1.
## 0 variants removed due to missing genotype data (--geno).
## --hwe: 0 variants removed due to Hardy-Weinberg exact test.
## 0 variants removed due to minor allele threshold(s)
## (--maf/--max-maf/--mac/--max-mac).
## 5 variants and 312 people pass filters and QC.
## Note: No phenotypes present.
## --recode ped to ./Data/QC/younameit.ped + ./Data/QC/younameit.map ... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99%done.</code></pre>
<p>The output files will be <strong>“younameit.map”</strong> and <strong>“younameit.ped”</strong> which have been quality controlled by the criteria you specified.</p>
</div>
<div id="remove-closely-related-individuals" class="section level2 hasAnchor" number="3.6">
<h2><span class="header-section-number">3.6</span> Remove Closely Related Individuals<a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#remove-closely-related-individuals" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Input files: in folder <code>.QC</code></p>
<ul>
<li><p>5SNPs.ped</p></li>
<li><p>5SNPs.map</p></li>
</ul>
<p>The following command will calculate <strong>indentical by descent (IBS) distances</strong> between all individuals.
1. <strong>–genome</strong>, calculate IBS</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb32-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb32-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb32-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb32-2" tabindex="-1"></a><span class="co"># modify the path to &quot;plink&quot; program</span></span>
<span id="cb32-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb32-3" tabindex="-1"></a></span>
<span id="cb32-4"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb32-4" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb32-5"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb32-5" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64/plink.exe</span> <span class="dt">\</span></span>
<span id="cb32-6"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb32-6" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--genome</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<div class="sourceCode" id="cb33"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb33-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb33-1" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb33-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb33-2" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac/plink</span> <span class="dt">\</span></span>
<span id="cb33-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb33-3" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/QC/5SNPs <span class="at">--genome</span> <span class="at">--out</span> ./Data/QC/younameit</span></code></pre></div>
<pre><code>## PLINK v1.90p 64-bit (11 Dec 2023)              www.cog-genomics.org/plink/1.9/
## (C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3
## Logging to ./Data/QC/younameit.log.
## Options in effect:
##   --file ./Data/QC/5SNPs
##   --genome
##   --noweb
##   --out ./Data/QC/younameit
## 
## Note: --noweb has no effect since no web check is implemented yet.
## 16384 MB RAM detected; reserving 8192 MB for main workspace.
## Scanning .ped file... 0%1%1%1%1%2%2%2%3%3%3%3%4%4%4%5%5%5%5%6%6%6%7%7%7%7%8%8%8%9%9%9%9%10%10%10%11%11%11%11%12%12%12%13%13%13%13%14%14%14%15%15%15%15%16%16%16%17%17%17%17%18%18%18%19%19%19%19%20%20%20%21%21%21%21%22%22%22%23%23%23%23%24%24%24%25%25%25%25%26%26%26%27%27%27%27%28%28%28%29%29%29%29%30%30%30%31%31%31%32%32%32%32%33%33%33%34%34%34%34%35%35%35%36%36%36%37%37%37%37%38%38%38%39%39%39%39%40%40%40%41%41%41%41%42%42%42%43%43%43%44%44%44%44%45%45%45%46%46%46%46%47%47%47%48%48%48%49%49%49%49%50%50%50%51%51%51%51%52%52%52%53%53%53%53%54%54%54%55%55%55%56%56%56%56%57%57%57%58%58%58%58%59%59%59%60%60%60%61%61%61%61%62%62%62%63%63%63%63%64%64%64%65%65%65%66%66%66%66%67%67%67%68%68%68%68%69%69%69%70%70%70%70%71%71%71%72%72%72%73%73%73%73%74%74%74%75%75%75%75%76%76%76%77%77%77%78%78%78%78%79%79%79%80%80%80%80%81%81%81%82%82%82%83%83%83%83%84%84%84%85%85%85%85%86%86%86%87%87%87%87%88%88%88%89%89%89%90%90%90%90%91%91%91%92%92%92%92%93%93%93%94%94%94%95%95%95%95%96%96%96%97%97%97%97%98%98%98%99%99%99%100%.ped scan complete (for binary autoconversion).
## Performing single-pass .bed write (5 variants, 344 people).
## 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%--file: ./Data/QC/younameit-temporary.bed + ./Data/QC/younameit-temporary.bim +
## ./Data/QC/younameit-temporary.fam written.
## 5 variants loaded from .bim file.
## 344 people (281 males, 0 females, 63 ambiguous) loaded from .fam.
## Ambiguous sex IDs written to ./Data/QC/younameit.nosex .
## Using up to 8 threads (change this with --threads).
## Before main variant filters, 344 founders and 0 nonfounders present.
## Calculating allele frequencies... 0%1%2%3%4%5%6%7%8%9%10%11%12%13%14%15%16%17%18%19%20%21%22%23%24%25%26%27%28%29%30%31%32%33%34%35%36%37%38%39%40%41%42%43%44%45%46%47%48%49%50%51%52%53%54%55%56%57%58%59%60%61%62%63%64%65%66%67%68%69%70%71%72%73%74%75%76%77%78%79%80%81%82%83%84%85%86%87%88%89%90%91%92%93%94%95%96%97%98%99% done.
## Total genotyping rate is 0.976744.
## 5 variants and 344 people pass filters and QC.
## Note: No phenotypes present.
## 5 markers complete.IBD calculations complete.  
## Writing... 1%Writing... 4%Writing... 6%Writing... 8%Writing... 11%Writing... 13%Writing... 15%Writing... 17%Writing... 20%Writing... 22%Writing... 24%Writing... 26%Writing... 29%Writing... 31%Writing... 33%Writing... 36%Writing... 38%Writing... 40%Writing... 42%Writing... 45%Writing... 47%Writing... 49%Writing... 51%Writing... 54%Writing... 56%Writing... 58%Writing... 61%Writing... 63%Writing... 65%Writing... 67%Writing... 70%Writing... 72%Writing... 74%Writing... 77%Writing... 79%Writing... 81%Writing... 83%Writing... 86%Writing... 88%Writing... 90%Writing... 92%Writing... 95%Writing... 97%Writing... 99%Finished writing ./Data/QC/younameit.genome .</code></pre>
<p>This will create the file <strong>younameit.genome</strong> which has the following fields:</p>
<pre><code> FID1      Family ID for first individual
 
 IID1      Individual ID for first individual
 
 FID2      Family ID for second individual
 
 IID2      Individual ID for second individual
 
 RT        Relationship type given PED file
 
 EZ        Expected IBD sharing given PED file
 
 Z0        P(IBD=0)
 
 Z1        P(IBD=1)
 
 Z2        P(IBD=2)
 
 PI_HAT    P(IBD=2)+0.5*P(IBD=1) ( proportion IBD )
 
 PHE       Pairwise phenotypic code (1,0,-1 = AA, AU and UU pairs)
 
 DST       IBS distance (IBS2 + 0.5*IBS1) / ( N SNP pairs )
 
 PPC       IBS binomial test
 
 RATIO     Of HETHET : IBS 0 SNPs (expected value is 2)</code></pre>
<p>Scan the output file <strong>“younameit.genome”</strong> for any individuals with high <strong>PI_HAT</strong> values (e.g. greater than 0.05).</p>
<p>Optionally, remove one member of the pair if close relatives are identified.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb36-1"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb36-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb36-2"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb36-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb36-3"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#cb36-3" tabindex="-1"></a><span class="fu">head</span> ./Data/QC/younameit.genome</span></code></pre></div>
<pre><code>##  FID1     IID1 FID2     IID2 RT    EZ      Z0      Z1      Z2  PI_HAT PHE       DST     PPC   RATIO
##    1  206-001   2  206-003 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   3  206-004 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   4  206-014 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   5  206-015 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA
##    1  206-001   6  206-016 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA
##    1  206-001   7  206-018 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001   8  206-019 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA
##    1  206-001   9  206-021 UN    NA  0.0000  0.0000  1.0000  1.0000  -1  1.000000      NA      NA
##    1  206-001  10  206-023 UN    NA  0.0000  1.0000  0.0000  0.5000  -1  0.600000      NA      NA</code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="an-overview-of-regression-with-toy-examples.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="an-imputation-example.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-gwastool.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_2024.pdf", "cshl_2024.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
