<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Simple Association Test | 2024 CSHL course - GWAS Hands-on Analysis</title>
  <meta name="description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  <meta name="generator" content="bookdown 0.39 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Simple Association Test | 2024 CSHL course - GWAS Hands-on Analysis" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  <meta name="github-repo" content="rstudio/cshl_2024" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Simple Association Test | 2024 CSHL course - GWAS Hands-on Analysis" />
  
  <meta name="twitter:description" content="This is the course material for the 2024 CSHL hands-on session in GWAS." />
  

<meta name="author" content="Danni Liu - Purdue University" />


<meta name="date" content="2024-06-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="population-stratification-correction-using-eigenstrat.html"/>
<link rel="next" href="unit-3.-gwas-case-study-continuous-trait.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">A Minimal Book Example</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a></li>
<li class="chapter" data-level="2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html"><i class="fa fa-check"></i><b>2</b> An Overview of Regression with Toy Examples</a>
<ul>
<li class="chapter" data-level="2.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#environment-setting-and-packages"><i class="fa fa-check"></i><b>2.1</b> Environment Setting and Packages</a></li>
<li class="chapter" data-level="2.2" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#toy-example-data"><i class="fa fa-check"></i><b>2.2</b> Toy Example Data</a></li>
<li class="chapter" data-level="2.3" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#logistic-regression"><i class="fa fa-check"></i><b>2.3</b> Logistic Regression</a></li>
<li class="chapter" data-level="2.4" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#roc-curve-and-auc-for-hoxb13"><i class="fa fa-check"></i><b>2.4</b> ROC curve and AUC for HOXB13</a></li>
<li class="chapter" data-level="2.5" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#practice"><i class="fa fa-check"></i><b>2.5</b> Practice</a></li>
<li class="chapter" data-level="2.6" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#comparison-of-hoxb13-roc-curve-and-nhlh1-roc-curve"><i class="fa fa-check"></i><b>2.6</b> Comparison of HOXB13 ROC curve and NHLH1 ROC curve</a></li>
<li class="chapter" data-level="2.7" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression"><i class="fa fa-check"></i><b>2.7</b> Linear Regression</a>
<ul>
<li class="chapter" data-level="2.7.1" data-path="an-overview-of-regression-with-toy-examples.html"><a href="an-overview-of-regression-with-toy-examples.html#linear-regression-of-hoxb13-vs.-il17br"><i class="fa fa-check"></i><b>2.7.1</b> Linear regression of HOXB13 vs. IL17BR</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><i class="fa fa-check"></i><b>3</b> GWAS Tools and Data Preprocessing Workflow with Toy Example – PLINK and IMPUTE2</a>
<ul>
<li class="chapter" data-level="3.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#environment-setting"><i class="fa fa-check"></i><b>3.1</b> Environment Setting</a></li>
<li class="chapter" data-level="3.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#gwas-tools"><i class="fa fa-check"></i><b>3.2</b> GWAS Tools</a></li>
<li class="chapter" data-level="3.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#quality-control"><i class="fa fa-check"></i><b>3.3</b> Quality Control</a></li>
<li class="chapter" data-level="3.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-summary-statistics"><i class="fa fa-check"></i><b>3.4</b> An Example of Summary Statistics</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#input-files"><i class="fa fa-check"></i><b>3.4.1</b> Input files</a></li>
<li class="chapter" data-level="3.4.2" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-missing-rate"><i class="fa fa-check"></i><b>3.4.2</b> Calculating missing rate</a></li>
<li class="chapter" data-level="3.4.3" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#calculating-allel-frequencies"><i class="fa fa-check"></i><b>3.4.3</b> Calculating allel frequencies</a></li>
<li class="chapter" data-level="3.4.4" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#hardy-weinberg-equilibrium"><i class="fa fa-check"></i><b>3.4.4</b> Hardy-Weinberg Equilibrium</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#an-example-of-inclusion-thresholds-for-qc"><i class="fa fa-check"></i><b>3.5</b> An Example of Inclusion Thresholds for QC</a></li>
<li class="chapter" data-level="3.6" data-path="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html"><a href="gwas-tools-and-data-preprocessing-workflow-with-toy-example-plink-and-impute2.html#remove-closely-related-individuals"><i class="fa fa-check"></i><b>3.6</b> Remove Closely Related Individuals</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="an-imputation-example.html"><a href="an-imputation-example.html"><i class="fa fa-check"></i><b>4</b> An Imputation Example</a></li>
<li class="chapter" data-level="5" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html"><i class="fa fa-check"></i><b>5</b> Population Stratification Correction Using EIGENSTRAT</a>
<ul>
<li class="chapter" data-level="5.1" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html#eigenstrat-input-files"><i class="fa fa-check"></i><b>5.1</b> EIGENSTRAT input files</a></li>
<li class="chapter" data-level="5.2" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html#run-eigenstrat-using-input-genotype-data"><i class="fa fa-check"></i><b>5.2</b> Run EIGENSTRAT Using Input Genotype Data</a></li>
<li class="chapter" data-level="5.3" data-path="population-stratification-correction-using-eigenstrat.html"><a href="population-stratification-correction-using-eigenstrat.html#test-the-significance-of-principle-components-pcs"><i class="fa fa-check"></i><b>5.3</b> Test the Significance of Principle Components (PCs)</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="simple-association-test.html"><a href="simple-association-test.html"><i class="fa fa-check"></i><b>6</b> Simple Association Test</a>
<ul>
<li class="chapter" data-level="6.1" data-path="simple-association-test.html"><a href="simple-association-test.html#quantitative-trait-association-test-for-continuous-trait"><i class="fa fa-check"></i><b>6.1</b> Quantitative Trait Association Test for Continuous Trait</a></li>
<li class="chapter" data-level="6.2" data-path="simple-association-test.html"><a href="simple-association-test.html#casecontrol-asssociation-test-for-binary-trait"><i class="fa fa-check"></i><b>6.2</b> Case/Control Asssociation Test for Binary Trait</a></li>
<li class="chapter" data-level="6.3" data-path="simple-association-test.html"><a href="simple-association-test.html#quantitative-trait-association-test-for-continuous-trait-with-covariates"><i class="fa fa-check"></i><b>6.3</b> Quantitative Trait Association Test for Continuous Trait with Covariates</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="simple-association-test.html"><a href="simple-association-test.html#covariates-adjustments-using-r"><i class="fa fa-check"></i><b>6.3.1</b> Covariates adjustments (using R):</a></li>
<li class="chapter" data-level="6.3.2" data-path="simple-association-test.html"><a href="simple-association-test.html#residual-phenotype-data-after-covariates-adjustments-in-unix"><i class="fa fa-check"></i><b>6.3.2</b> Residual phenotype data after covariates adjustments (in UNIX):</a></li>
<li class="chapter" data-level="6.3.3" data-path="simple-association-test.html"><a href="simple-association-test.html#association-test-using-plink"><i class="fa fa-check"></i><b>6.3.3</b> Association test using plink</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html"><i class="fa fa-check"></i><b>7</b> Unit 3. GWAS Case Study-Continuous Trait</a>
<ul>
<li class="chapter" data-level="7.1" data-path="unit-3.-gwas-case-study-continuous-trait.html"><a href="unit-3.-gwas-case-study-continuous-trait.html#data-preprocessing"><i class="fa fa-check"></i><b>7.1</b> Data Preprocessing</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="applications.html"><a href="applications.html"><i class="fa fa-check"></i><b>8</b> Applications</a>
<ul>
<li class="chapter" data-level="8.1" data-path="applications.html"><a href="applications.html#example-one"><i class="fa fa-check"></i><b>8.1</b> Example one</a></li>
<li class="chapter" data-level="8.2" data-path="applications.html"><a href="applications.html#example-two"><i class="fa fa-check"></i><b>8.2</b> Example two</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>9</b> Final Words</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">2024 CSHL course - GWAS Hands-on Analysis</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="simple-association-test" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Simple Association Test<a href="simple-association-test.html#simple-association-test" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>PLINK is a free, open-source whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner, see <a href="http://zzz.bwh.harvard.edu/plink/" class="uri">http://zzz.bwh.harvard.edu/plink/</a>.</p>
<p>The input files include PED file, MAP file, alternate phenotype file, and optional covariate file.</p>
<p>The PED file is a white-space (space or tab) delimited file, the first six columns are mandatory:</p>
<pre><code>Col 1: Family ID 

Col 2: Individual ID 

Col 3: Paternal ID 

Col 4: Maternal ID 

Col 5: Sex (1=male; 2=female; other character =unknown) 

Col 6: Phenotype (The missing phenotype value for quantitative traits is, by default, -9)

Col 7(*): Genotype (The missing genotype value is denoted as 0, by default)</code></pre>
<p>Example:</p>
<pre><code>FAM001 1 0 0 1 3.4 A A G G A C C C 

FAM001 2 0 0 1 2.5 A A A G 0 0 A C</code></pre>
<p>The MAP file has four columns:</p>
<pre><code>Col 1: Chromosome (1-22, X, Y or 0 if unplaced)

Col 2: rs# or SNP identifier 

Col 3: Genetic distance (morgans) 

Col 4: Base-pair position (bp units)</code></pre>
<p>To specify an alternate phenotype for analysis, i.e. other than the one in the .ped file, use the option –pheno. The alternate phenotype file has three columns:</p>
<pre><code> Col 1: Family ID
 
 Col 2: Individual ID
 
 Col 3: Phenotype</code></pre>
<p>Example:</p>
<pre><code> F1     1110     2.30   22.22  2  
 
 F2     2202     34.12  18.23  1  
 </code></pre>
<p>The phenotype can be either a quantitative trait or an affection status column. PLINK will automatically detect which type it is based on whether a value other than 0, 1, 2 or the missing genotype code is observed.</p>
<p>(All files and results in this section are saved in <code>./PC</code> folder.)</p>
<div id="quantitative-trait-association-test-for-continuous-trait" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Quantitative Trait Association Test for Continuous Trait<a href="simple-association-test.html#quantitative-trait-association-test-for-continuous-trait" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below is an example that demonstrates how to use PLINK to perform a quantitative trait association test for a toy dataset with a continuous trait and 5 SNPs:</p>
<ul>
<li>Genotype data:
<ul>
<li>5SNPs.ped</li>
<li>5SNPs.map</li>
</ul></li>
<li>Phenotype data:
<ul>
<li>conty.txt</li>
</ul></li>
</ul>
<p>The following commands perform the <strong>simple association test</strong></p>
<ol style="list-style-type: decimal">
<li><strong>–assoc</strong>, perform case/control or QTL association. You can also use <strong>–fisher</strong> to do Fisher’s exact (allelic) test or <strong>–model</strong> to do Cochran-Armitage and full-model C/C association.</li>
<li><strong>–pheno</strong>, specify the phenotype file.</li>
</ol>
<div class="sourceCode" id="cb44"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb44-1"><a href="simple-association-test.html#cb44-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb44-2"><a href="simple-association-test.html#cb44-2" tabindex="-1"></a></span>
<span id="cb44-3"><a href="simple-association-test.html#cb44-3" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb44-4"><a href="simple-association-test.html#cb44-4" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20230116/plink</span> <span class="dt">\</span></span>
<span id="cb44-5"><a href="simple-association-test.html#cb44-5" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/conty.txt <span class="at">--out</span> ./Data/Assoc/younameit.conty</span>
<span id="cb44-6"><a href="simple-association-test.html#cb44-6" tabindex="-1"></a></span>
<span id="cb44-7"><a href="simple-association-test.html#cb44-7" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb44-8"><a href="simple-association-test.html#cb44-8" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20230116/plink.exe</span> <span class="dt">\</span></span>
<span id="cb44-9"><a href="simple-association-test.html#cb44-9" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/conty.txt <span class="at">--out</span> ./Data/Assoc/younameit.conty</span></code></pre></div>
<p>This will generate the output file <strong>“younameit.conty.qassoc”</strong> with following columns:</p>
<pre><code>CHR      Chromosome number

SNP      SNP identifier

BP       Physical position (base-pair)

NMISS    Number of non-missing genotypes

BETA     Regression coefficient

SE       Standard error

R2       Regression r-squared

T        Wald test (based on t-distribution)

P        Wald test asymptotic p-value</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb46-1"><a href="simple-association-test.html#cb46-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb46-2"><a href="simple-association-test.html#cb46-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb46-3"><a href="simple-association-test.html#cb46-3" tabindex="-1"></a><span class="fu">head</span> ./Data/Assoc/younameit.conty.qassoc</span></code></pre></div>
<pre><code>##  CHR         SNP         BP    NMISS       BETA         SE         R2        T            P 
##    1   rs3094315     752566      280    0.07487     0.1485  0.0009132   0.5041       0.6146 
##    1   rs2073813     753541      272    0.07533     0.1007   0.002068   0.7481       0.4551 
##    1   rs3131969     754182      271     0.1038    0.09773   0.004176    1.062       0.2892 
##    1   rs3131967     754334      280    0.06321     0.1045   0.001313   0.6046        0.546 
##    1  rs77598327     757120      275    0.08091     0.1341   0.001332   0.6035       0.5467</code></pre>
</div>
<div id="casecontrol-asssociation-test-for-binary-trait" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Case/Control Asssociation Test for Binary Trait<a href="simple-association-test.html#casecontrol-asssociation-test-for-binary-trait" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below is an example that demonstrates how to use PLINK to perform a case/control association test for a binary trait and 5 SNPs:</p>
<ul>
<li><p>Genotype data:</p>
<ul>
<li><p>5SNPs.ped</p></li>
<li><p>5SNPs.map</p></li>
</ul></li>
<li><p>Phenotype data:</p>
<ul>
<li>biny.txt</li>
</ul></li>
</ul>
<p>The following commands perform the <strong>simple association test</strong></p>
<ol style="list-style-type: decimal">
<li><strong>–assoc</strong>, perform case/control or QTL association. You can also use <strong>–fisher</strong> to do Fisher’s exact (allelic) test or <strong>–model</strong> to do Cochran-Armitage and full-model C/C association.</li>
<li><strong>–pheno</strong>, specify the phenotype file.</li>
</ol>
<div class="sourceCode" id="cb48"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb48-1"><a href="simple-association-test.html#cb48-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb48-2"><a href="simple-association-test.html#cb48-2" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb48-3"><a href="simple-association-test.html#cb48-3" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20230116/plink</span> <span class="dt">\</span></span>
<span id="cb48-4"><a href="simple-association-test.html#cb48-4" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/biny.txt <span class="at">--out</span> ./Data/Assoc/younameit.biny</span>
<span id="cb48-5"><a href="simple-association-test.html#cb48-5" tabindex="-1"></a></span>
<span id="cb48-6"><a href="simple-association-test.html#cb48-6" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb48-7"><a href="simple-association-test.html#cb48-7" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20230116/plink.exe</span> <span class="dt">\</span></span>
<span id="cb48-8"><a href="simple-association-test.html#cb48-8" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/biny.txt <span class="at">--out</span> ./Data/Assoc/younameit.biny</span></code></pre></div>
<p>The output is <strong>“younameit.biny.assoc”</strong>, which contains the fields:</p>
<pre><code> CHR     Chromosome
 SNP     SNP ID
 BP      Physical position (base-pair)
 A1      Minor allele name (based on whole sample)
 F_A     Frequency of this allele in cases
 F_U     Frequency of this allele in controls
 A2      Major allele name
 CHISQ   Basic allelic test chi-square (1df)
 P       Asymptotic p-value for this test
 OR      Estimated odds ratio (for A1, i.e. A2 is reference)</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb50-1"><a href="simple-association-test.html#cb50-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb50-2"><a href="simple-association-test.html#cb50-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb50-3"><a href="simple-association-test.html#cb50-3" tabindex="-1"></a><span class="fu">head</span> ./Data/Assoc/younameit.biny.assoc</span></code></pre></div>
<pre><code>##  CHR         SNP         BP   A1      F_A      F_U   A2        CHISQ            P           OR 
##    1   rs3094315     752566    G  0.07664  0.09091    A       0.3705       0.5428         0.83 
##    1   rs2073813     753541    A   0.2239   0.2391    G       0.1777       0.6734       0.9178 
##    1   rs3131969     754182    A   0.2176   0.2607    G        1.383       0.2396       0.7884 
##    1   rs3131967     754334    A   0.1949   0.2326    G        1.186       0.2761       0.7983 
##    1  rs77598327     757120    G   0.1111   0.1393    A       0.9945       0.3186       0.7724</code></pre>
</div>
<div id="quantitative-trait-association-test-for-continuous-trait-with-covariates" class="section level2 hasAnchor" number="6.3">
<h2><span class="header-section-number">6.3</span> Quantitative Trait Association Test for Continuous Trait with Covariates<a href="simple-association-test.html#quantitative-trait-association-test-for-continuous-trait-with-covariates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Below is an example that demonstrates how to use PLINK to perform a quantitative trait test for a continuous trait and 5 SNPs with 3 covariates:</p>
<ol style="list-style-type: decimal">
<li>Genotype data:</li>
</ol>
<ul>
<li>5SNPs.ped</li>
<li>5SNPs.map</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Phenotype data:</li>
</ol>
<ul>
<li>conty_pheno.txt</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Covariates:</li>
</ol>
<ul>
<li>conty_covar.txt</li>
</ul>
<p>Before the association test, we adjust the phenotypes to remove the effects of covariates. For the continous phenotype, we employ linear regression to adjust the phenotype. Then, we use the residual as the new phenotype in the association test.</p>
<div id="covariates-adjustments-using-r" class="section level3 hasAnchor" number="6.3.1">
<h3><span class="header-section-number">6.3.1</span> Covariates adjustments (using R):<a href="simple-association-test.html#covariates-adjustments-using-r" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="simple-association-test.html#cb52-1" tabindex="-1"></a><span class="do">## R codes- run in R Console</span></span>
<span id="cb52-2"><a href="simple-association-test.html#cb52-2" tabindex="-1"></a><span class="co"># Read phenotype data and covariates data</span></span>
<span id="cb52-3"><a href="simple-association-test.html#cb52-3" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/Assoc/conty_pheno.txt&quot;</span>) <span class="co">#read phenotype data</span></span>
<span id="cb52-4"><a href="simple-association-test.html#cb52-4" tabindex="-1"></a>covar<span class="ot">=</span><span class="fu">read.table</span>(<span class="st">&quot;./Data/Assoc/conty_covar.txt&quot;</span>) <span class="co">#read covariates data</span></span>
<span id="cb52-5"><a href="simple-association-test.html#cb52-5" tabindex="-1"></a>pheno<span class="ot">=</span><span class="fu">as.matrix</span>(pheno) <span class="co">#create a matrix from the phenotype data </span></span>
<span id="cb52-6"><a href="simple-association-test.html#cb52-6" tabindex="-1"></a>covar<span class="ot">=</span><span class="fu">as.matrix</span>(covar) <span class="co">#create a matrix from the covariates data</span></span>
<span id="cb52-7"><a href="simple-association-test.html#cb52-7" tabindex="-1"></a>n<span class="ot">=</span><span class="fu">dim</span>(pheno)[<span class="dv">1</span>] <span class="co">#sample size</span></span>
<span id="cb52-8"><a href="simple-association-test.html#cb52-8" tabindex="-1"></a>p<span class="ot">=</span><span class="fu">dim</span>(pheno)[<span class="dv">2</span>] <span class="co">#number of traits</span></span></code></pre></div>
<p>Covariates adjustment is done by fitting a linear regression model and obtain residuals as new phenotype data for association test.</p>
<p>The following codes construct the linear model for each of the traits using <code>lm</code> function.The resulted residual data is saved for analysis.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="simple-association-test.html#cb53-1" tabindex="-1"></a><span class="co"># R codes- run in R Console</span></span>
<span id="cb53-2"><a href="simple-association-test.html#cb53-2" tabindex="-1"></a>fit<span class="ot">=</span><span class="fu">list</span>() <span class="co">#create an empty list</span></span>
<span id="cb53-3"><a href="simple-association-test.html#cb53-3" tabindex="-1"></a>residpheno<span class="ot">=</span><span class="fu">matrix</span>(<span class="dv">0</span>, n, p) <span class="co">#create an empty matrix</span></span>
<span id="cb53-4"><a href="simple-association-test.html#cb53-4" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>p){</span>
<span id="cb53-5"><a href="simple-association-test.html#cb53-5" tabindex="-1"></a>  fit[[i]]<span class="ot">=</span><span class="fu">lm</span>(pheno[,i]<span class="sc">~</span>covar) <span class="co">#for each trait, fit a linear regression model of phenotype against covariates</span></span>
<span id="cb53-6"><a href="simple-association-test.html#cb53-6" tabindex="-1"></a>  residpheno[,i]<span class="ot">=</span><span class="fu">resid</span>(fit[[i]]) <span class="co">#obtain residual phenotype data</span></span>
<span id="cb53-7"><a href="simple-association-test.html#cb53-7" tabindex="-1"></a>}</span>
<span id="cb53-8"><a href="simple-association-test.html#cb53-8" tabindex="-1"></a><span class="fu">write.table</span>(residpheno, <span class="st">&quot;./Data/Assoc/resid_phenotype.txt&quot;</span>, <span class="at">row.names=</span>F, <span class="at">col.names=</span>F, <span class="at">quote=</span>F, <span class="at">sep=</span><span class="st">&quot; &quot;</span>) <span class="co">#save residual phenotype data</span></span></code></pre></div>
</div>
<div id="residual-phenotype-data-after-covariates-adjustments-in-unix" class="section level3 hasAnchor" number="6.3.2">
<h3><span class="header-section-number">6.3.2</span> Residual phenotype data after covariates adjustments (in UNIX):<a href="simple-association-test.html#residual-phenotype-data-after-covariates-adjustments-in-unix" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This code combines the above residual data and sample ID information to contruct the phenotype data for PLINK.</p>
<div class="sourceCode" id="cb54"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb54-1"><a href="simple-association-test.html#cb54-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb54-2"><a href="simple-association-test.html#cb54-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb54-3"><a href="simple-association-test.html#cb54-3" tabindex="-1"></a><span class="fu">paste</span> ./Data/Assoc/famID.txt ./Data/Assoc/sampleID.txt <span class="dt">\</span></span>
<span id="cb54-4"><a href="simple-association-test.html#cb54-4" tabindex="-1"></a>./Data/Assoc/resid_phenotype.txt <span class="op">&gt;</span> ./Data/Assoc/resid_phenotype_assoc.txt</span></code></pre></div>
<p>Then we can use PLINK to perform the association test.</p>
</div>
<div id="association-test-using-plink" class="section level3 hasAnchor" number="6.3.3">
<h3><span class="header-section-number">6.3.3</span> Association test using plink<a href="simple-association-test.html#association-test-using-plink" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After adjusting the phenotype using a set of covariates, we use the residual as new phenotype for the association test. <code>PLINK</code> was used again to perform assiciation test.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb55-1"><a href="simple-association-test.html#cb55-1" tabindex="-1"></a><span class="co"># both Mac and Windows can run in Terminal</span></span>
<span id="cb55-2"><a href="simple-association-test.html#cb55-2" tabindex="-1"></a></span>
<span id="cb55-3"><a href="simple-association-test.html#cb55-3" tabindex="-1"></a><span class="co"># macOS user</span></span>
<span id="cb55-4"><a href="simple-association-test.html#cb55-4" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_mac_20230116/plink</span> <span class="dt">\</span></span>
<span id="cb55-5"><a href="simple-association-test.html#cb55-5" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/resid_phenotype_assoc.txt <span class="dt">\</span></span>
<span id="cb55-6"><a href="simple-association-test.html#cb55-6" tabindex="-1"></a>--out ./Data/Assoc/younameit_resid</span>
<span id="cb55-7"><a href="simple-association-test.html#cb55-7" tabindex="-1"></a></span>
<span id="cb55-8"><a href="simple-association-test.html#cb55-8" tabindex="-1"></a><span class="co"># Windows user</span></span>
<span id="cb55-9"><a href="simple-association-test.html#cb55-9" tabindex="-1"></a><span class="ex">./Tools/PLINK/plink_win64_20230116/plink.exe</span> <span class="dt">\</span></span>
<span id="cb55-10"><a href="simple-association-test.html#cb55-10" tabindex="-1"></a>--noweb <span class="at">--file</span> ./Data/Assoc/5SNPs <span class="at">--assoc</span> <span class="at">--pheno</span> ./Data/Assoc/resid_phenotype_assoc.txt <span class="dt">\</span></span>
<span id="cb55-11"><a href="simple-association-test.html#cb55-11" tabindex="-1"></a>--out ./Data/Assoc/younameit_resid</span></code></pre></div>
<div class="sourceCode" id="cb56"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb56-1"><a href="simple-association-test.html#cb56-1" tabindex="-1"></a><span class="co"># bash codes- run in terminal</span></span>
<span id="cb56-2"><a href="simple-association-test.html#cb56-2" tabindex="-1"></a><span class="co"># Windows user need to run in Powershell App</span></span>
<span id="cb56-3"><a href="simple-association-test.html#cb56-3" tabindex="-1"></a><span class="co">### check association results</span></span>
<span id="cb56-4"><a href="simple-association-test.html#cb56-4" tabindex="-1"></a><span class="fu">head</span> ./Data/Assoc/younameit_resid.qassoc</span></code></pre></div>
<pre><code>##  CHR         SNP         BP    NMISS       BETA         SE         R2        T            P 
##    1   rs3094315     752566      280     0.0706      0.148  0.0008178    0.477       0.6337 
##    1   rs2073813     753541      272    0.07739     0.1003   0.002198   0.7713       0.4412 
##    1   rs3131969     754182      271     0.1062     0.0974     0.0044     1.09       0.2766 
##    1   rs3131967     754334      280    0.06205     0.1042   0.001274   0.5954       0.5521 
##    1  rs77598327     757120      275     0.0816     0.1336   0.001366    0.611       0.5417</code></pre>
<p>This output file <strong>“younameit.qassoc”</strong> has the following columns:</p>
<pre><code>CHR      Chromosome number

SNP      SNP identifier

BP       Physical position (base-pair)

NMISS    Number of non-missing genotypes

BETA     Regression coefficient

SE       Standard error

R2       Regression r-squared

T        Wald test (based on t-distribution)

P        Wald test asymptotic p-value</code></pre>
<p>In the next section, we will start analyzing real data from a case study. The procedures might not be exactly the same here, as real data require more proprocessing steps before it can be used for association study.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="population-stratification-correction-using-eigenstrat.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="unit-3.-gwas-case-study-continuous-trait.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/02-gwastool.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["cshl_2024.pdf", "cshl_2024.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
